--- a/third_party/llvm/setup.bzl	2026-02-07 21:50:02.940838334 +0300
+++ b/third_party/llvm/setup.bzl	2026-02-07 21:50:36.657067040 +0300
@@ -4,15 +4,8 @@
 
 # The subset of LLVM targets that TensorFlow cares about.
 _LLVM_TARGETS = [
-    "AArch64",
-    "AMDGPU",
-    "ARM",
     "NVPTX",
-    "PowerPC",
-    "RISCV",
-    "SystemZ",
     "X86",
-    "SPIRV",
 ]
 
 def llvm_setup(name):
--- a/workspace2.bzl	2026-02-17 17:38:07.950956100 +0300
+++ b/workspace2.bzl	2026-02-17 17:40:43.233198682 +0300
@@ -435,11 +435,10 @@
 
     tf_http_archive(
         name = "boringssl",
-        sha256 = "9dc53f851107eaf87b391136d13b815df97ec8f76dadb487b58b2fc45e624d2c",
-        strip_prefix = "boringssl-c00d7ca810e93780bd0c8ee4eea28f4f2ea4bcdc",
+        sha256 = "9fd4258b6c530059c1b85192c940f3ae5f5f9f0cb4213ce77f892f3eef62437c",
+        strip_prefix = "boringssl-617634bc015344093f5ea0b0a5b653c924cfa20d",
         system_build_file = "//third_party:boringssl.BUILD",
-        patch_file = ["//third_party:boringssl.patch"],
-        urls = tf_mirror_urls("https://github.com/google/boringssl/archive/c00d7ca810e93780bd0c8ee4eea28f4f2ea4bcdc.tar.gz"),
+        urls = tf_mirror_urls("https://github.com/google/boringssl/archive/617634bc015344093f5ea0b0a5b653c924cfa20d.tar.gz"),
     )
 
     tf_http_archive(
