diff -wbBur Tix8.4.3/configure Tix8.4.3.my/configure
--- Tix8.4.3/configure	2008-02-28 04:35:01.000000000 +0000
+++ Tix8.4.3.my/configure	2009-12-28 10:57:33.000000000 +0000
@@ -6645,7 +6645,7 @@
 	        ;;
 	esac
     else
-	if test ! -f "${TCL_SRC_DIR}/generic/tclInt.h" ; then
+	if test ! -f "${TCL_SRC_DIR}/tclInt.h" ; then
 	    { { echo "$as_me:$LINENO: error: Cannot find private header tclInt.h in ${TCL_SRC_DIR}" >&5
 echo "$as_me: error: Cannot find private header tclInt.h in ${TCL_SRC_DIR}" >&2;}
    { (exit 1); exit 1; }; }
@@ -6700,7 +6700,7 @@
 	        ;;
 	esac
     else
-	if test ! -f "${TK_SRC_DIR}/generic/tkInt.h" ; then
+	if test ! -f "${TK_SRC_DIR}/tk-private/generic/tkInt.h" ; then
 	    { { echo "$as_me:$LINENO: error: Cannot find private header tkInt.h in ${TK_SRC_DIR}" >&5
 echo "$as_me: error: Cannot find private header tkInt.h in ${TK_SRC_DIR}" >&2;}
    { (exit 1); exit 1; }; }

