diff --git a/3rdparty/civetweb/civetweb.cmake b/3rdparty/civetweb/civetweb.cmake
index 5d4749fe4..15b1f9720 100644
--- a/3rdparty/civetweb/civetweb.cmake
+++ b/3rdparty/civetweb/civetweb.cmake
@@ -17,6 +17,7 @@ ExternalProject_Add(
         -DCIVETWEB_ENABLE_DEBUG_TOOLS=OFF
         -DCIVETWEB_DISABLE_CGI=ON
         -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DCMAKE_CXX_FLAGS="-D_GLIBCXX_USE_CXX11_ABI=$<BOOL:${GLIBCXX_USE_CXX11_ABI}> $<$<PLATFORM_ID:Windows>:/EHsc>"
         ${ExternalProject_CMAKE_ARGS_hidden}
     BUILD_BYPRODUCTS
