diff --git a/3rdparty/librealsense/librealsense.cmake b/3rdparty/librealsense/librealsense.cmake
index e5caa700d..42dfe16a6 100644
--- a/3rdparty/librealsense/librealsense.cmake
+++ b/3rdparty/librealsense/librealsense.cmake
@@ -30,6 +30,7 @@ ExternalProject_Add(
         -DBUILD_PYTHON_BINDINGS=OFF
         -DBUILD_WITH_CUDA=${BUILD_CUDA_MODULE}
         -DUSE_EXTERNAL_USB=ON
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         # Syncing GLIBCXX_USE_CXX11_ABI for MSVC causes problems, but directly
         # checking CXX_COMPILER_ID is not supported.
         $<IF:$<PLATFORM_ID:Windows>,"",-DCMAKE_CXX_FLAGS=-D_GLIBCXX_USE_CXX11_ABI=$<BOOL:${GLIBCXX_USE_CXX11_ABI}>>
