--- pyglossary-3.0.4/pyglossary/core.py.orig	2017-01-04 20:45:07.786188210 +0300
+++ pyglossary-3.0.4/pyglossary/core.py	2017-01-04 20:38:33.000000000 +0300
@@ -50,11 +50,10 @@
 
 if hasattr(sys, 'frozen'):
    rootDir = dirname(sys.executable)
-   uiDir = join(rootDir, 'ui')
 else:
-   uiDir = dirname(realpath(__file__))
-   rootDir = dirname(uiDir)
+   rootDir = dirname(realpath(sys.argv[0]))
 
+uiDir = join(rootDir, 'ui')
 resDir = join(rootDir, 'res')
 
 if os.sep=='/': ## Operating system is Unix-Like
