--- pyglossary-3.2.1/pyglossary/core.py.orig	2019-06-27 00:09:04.322675748 +0300
+++ pyglossary-3.2.1/pyglossary/core.py	2019-06-27 00:10:06.833397976 +0300
@@ -164,10 +164,10 @@
 
 if hasattr(sys, "frozen"):
 	rootDir = dirname(sys.executable)
-	uiDir = join(rootDir, "ui")
 else:
-	uiDir = dirname(realpath(__file__))
-	rootDir = dirname(uiDir)
+	rootDir = dirname(realpath(sys.argv[0]))
+
+uiDir = join(rootDir, 'ui')
 
 dataDir = rootDir
 if dataDir.endswith("dist-packages") or dataDir.endswith("site-packages"):
