diff --unified --recursive --text hipBLAS-rocm-3.0/library/CMakeLists.txt hipBLAS-rocm-3.0/library/CMakeLists.txt
--- hipBLAS-rocm-3.0/library/CMakeLists.txt	2019-12-07 01:18:42.000000000 +0100
+++ hipBLAS-rocm-3.0/library/CMakeLists.txt	2020-02-01 14:57:05.980017802 +0100
@@ -77,6 +77,7 @@
 
 set( HIPBLAS_CONFIG_DIR "\${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" CACHE PATH "Path placed into ldconfig file" )
 
+
 rocm_create_package(
     NAME ${package_name}
     DESCRIPTION "Radeon Open Compute BLAS marshalling library"
diff --unified --recursive --text hipBLAS-rocm-3.0/library/src/CMakeLists.txt hipBLAS-rocm-3.0/library/src/CMakeLists.txt
--- hipBLAS-rocm-3.0/library/src/CMakeLists.txt	2019-12-07 01:18:42.000000000 +0100
+++ hipBLAS-rocm-3.0/library/src/CMakeLists.txt	2020-02-01 14:57:13.314001035 +0100
@@ -87,6 +87,8 @@
 set_target_properties( hipblas PROPERTIES CXX_VISIBILITY_PRESET "hidden" VISIBILITY_INLINES_HIDDEN ON )
 generate_export_header( hipblas EXPORT_FILE_NAME ${PROJECT_BINARY_DIR}/include/hipblas-export.h )
 
+include_directories("/opt/rocm/hsa/include")
+
 # Following Boost conventions of prefixing 'lib' on static built libraries, across all platforms
 if( NOT BUILD_SHARED_LIBS )
   set_target_properties( hipblas PROPERTIES PREFIX "lib" )
