diff --git a/cmake/Modules/FindNNPACK.cmake b/cmake/Modules/FindNNPACK.cmake
index 547bf88f..29cf5dce 100644
--- a/cmake/Modules/FindNNPACK.cmake
+++ b/cmake/Modules/FindNNPACK.cmake
@@ -21,12 +21,17 @@ find_library(NNPACK_LIBRARY nnpack
     PATHS ${NNPACK_ROOT_DIR}
     PATH_SUFFIXES lib lib64)
 
-find_package_handle_standard_args(NNPACK DEFAULT_MSG NNPACK_INCLUDE_DIR NNPACK_LIBRARY)
+find_library(PTHREADPOOL_LIBRARY pthreadpool
+    PATHS ${NNPACK_ROOT_DIR}
+    PATH_SUFFIXES lib lib64)
+
+find_package_handle_standard_args(NNPACK DEFAULT_MSG NNPACK_INCLUDE_DIR NNPACK_LIBRARY PTHREADPOOL_LIBRARY)
 
 if(NNPACK_FOUND)
   set(NNPACK_INCLUDE_DIRS ${NNPACK_INCLUDE_DIR})
-  set(NNPACK_LIBRARIES ${NNPACK_LIBRARY})
+  set(NNPACK_LIBRARIES ${NNPACK_LIBRARY} ${PTHREADPOOL_LIBRARY})
   message(STATUS "Found NNPACK    (include: ${NNPACK_INCLUDE_DIR}, library: ${NNPACK_LIBRARY})")
+  message(STATUS "Found PTHREADPOOL (library: ${PTHREADPOOL_LIBRARY})")
   mark_as_advanced(NNPACK_ROOT_DIR NNPACK_LIBRARY_RELEASE NNPACK_LIBRARY_DEBUG
                                  NNPACK_LIBRARY NNPACK_INCLUDE_DIR)
 endif()
