diff --color -urt Trilinos-trilinos-release-14-0-0.orig/packages/krino/cmake/Dependencies.cmake Trilinos-trilinos-release-14-0-0.new/packages/krino/cmake/Dependencies.cmake
--- Trilinos-trilinos-release-14-0-0.orig/packages/krino/cmake/Dependencies.cmake	2023-05-18 08:34:11.085906271 +0200
+++ Trilinos-trilinos-release-14-0-0.new/packages/krino/cmake/Dependencies.cmake	2023-05-18 08:29:48.930168370 +0200
@@ -1,6 +1,6 @@
 SET(LIB_REQUIRED_DEP_PACKAGES SEACASIoss SEACASExodus SEACASAprepro STKBalance STKMath STKIO STKSearch STKTopology STKUtil STKTools STKEmend Percept Intrepid)
 SET(LIB_OPTIONAL_DEP_PACKAGES)
-SET(TEST_REQUIRED_DEP_PACKAGES Gtest STKUnit_test_utils)
+SET(TEST_REQUIRED_DEP_PACKAGES Gtest)
 SET(TEST_OPTIONAL_DEP_PACKAGES)
 SET(LIB_REQUIRED_DEP_TPLS Boost)
 SET(LIB_OPTIONAL_DEP_TPLS MPI ParMETIS yaml-cpp)
diff --color -urt Trilinos-trilinos-release-14-0-0.orig/packages/krino/krino/CMakeLists.txt Trilinos-trilinos-release-14-0-0.new/packages/krino/krino/CMakeLists.txt
--- Trilinos-trilinos-release-14-0-0.orig/packages/krino/krino/CMakeLists.txt	2023-05-18 08:34:11.085906271 +0200
+++ Trilinos-trilinos-release-14-0-0.new/packages/krino/krino/CMakeLists.txt	2023-05-18 08:35:25.202277721 +0200
@@ -4,7 +4,6 @@
 add_subdirectory(region)
 add_subdirectory(rebalance_utils)
 add_subdirectory(parser)
-add_subdirectory(unit_tests)
 
 SET(SOURCES_MAIN Apps_krino.cpp)
 
diff --color -urt Trilinos-trilinos-release-14-0-0.orig/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp Trilinos-trilinos-release-14-0-0.new/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp
--- Trilinos-trilinos-release-14-0-0.orig/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp	2023-05-18 08:34:11.735903734 +0200
+++ Trilinos-trilinos-release-14-0-0.new/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp	2023-05-18 08:29:49.596832958 +0200
@@ -43,7 +43,7 @@
 #define TEUCHOS_BIG_UINT_DECL_HPP
 
 #include <iosfwd>
-
+#include <cstdint>
 /*! \file Teuchos_BigUIntDecl.hpp
     \brief Arbitrary-precision unsigned integer declaration.
 */
