diff --color --color -uraBN brlcad-rel-7-42-0/db/nist/CMakeLists.txt brlcad-rel-7-42-0-patched/db/nist/CMakeLists.txt
--- brlcad-rel-7-42-0/db/nist/CMakeLists.txt	2025-08-11 15:50:22.000000000 -0500
+++ brlcad-rel-7-42-0-patched/db/nist/CMakeLists.txt	2025-12-31 14:28:56.611312350 -0500
@@ -11,7 +11,8 @@
 )
 
 foreach(im ${NIST_SAMPLE_MODELS})
-  g_target(${im} "BRL-CAD Geometry Models/Sample")
+  # disabled because this often segfaults
+  # g_target(${im} "BRL-CAD Geometry Models/Sample")
 endforeach(im ${NIST_SAMPLE_MODELS})
 
 cmakefiles(
