--- a/jaxlib/cuda/BUILD	2026-02-07 16:54:55.357216204 +0300
+++ b/jaxlib/cuda/BUILD	2026-02-07 16:55:32.146889315 +0300
@@ -76,6 +76,8 @@
     name = "cuda_make_batch_pointers",
     srcs = ["//jaxlib/gpu:make_batch_pointers.cu.cc"],
     hdrs = ["//jaxlib/gpu:make_batch_pointers.h"],
+    copts = ["-U_FORTIFY_SOURCE", "-D_FORTIFY_SOURCE=0"],
+    cxxopts = ["-U_FORTIFY_SOURCE", "-D_FORTIFY_SOURCE=0"],
     deps = [
         ":cuda_vendor",
         "@local_config_cuda//cuda:cuda_headers",
@@ -288,6 +290,8 @@
     ],
     hdrs = ["//jaxlib/gpu:linalg_kernels.h"],
     features = ["-use_header_modules"],
+    copts = ["-U_FORTIFY_SOURCE", "-D_FORTIFY_SOURCE=0"],
+    cxxopts = ["-U_FORTIFY_SOURCE", "-D_FORTIFY_SOURCE=0"],
     deps = [
         ":cuda_blas_handle_pool",
         ":cuda_gpu_kernel_helpers",
@@ -305,6 +309,8 @@
     name = "cuda_linalg_kernels_impl",
     srcs = ["//jaxlib/gpu:linalg_kernels.cu.cc"],
     hdrs = ["//jaxlib/gpu:linalg_kernels.h"],
+    copts = ["-U_FORTIFY_SOURCE", "-D_FORTIFY_SOURCE=0"],
+    cxxopts = ["-U_FORTIFY_SOURCE", "-D_FORTIFY_SOURCE=0"],
     deps = [
         ":cuda_gpu_kernel_helpers",
         ":cuda_vendor",
@@ -359,6 +365,8 @@
         "//jaxlib/gpu:prng_kernels.cu.cc",
     ],
     hdrs = ["//jaxlib/gpu:prng_kernels.h"],
+    copts = ["-U_FORTIFY_SOURCE", "-D_FORTIFY_SOURCE=0"],
+    cxxopts = ["-U_FORTIFY_SOURCE", "-D_FORTIFY_SOURCE=0"],
     deps = [
         ":cuda_gpu_kernel_helpers",
         ":cuda_vendor",
