diff --git a/src/julia_internal.h b/src/julia_internal.h
index 186b5d7b2b..33143f2062 100644
--- a/src/julia_internal.h
+++ b/src/julia_internal.h
@@ -623,12 +623,9 @@ extern volatile int jl_in_stackwalk;
 #  include <libunwind.h>
 typedef unw_context_t bt_context_t;
 typedef unw_cursor_t bt_cursor_t;
-#  if (!defined(SYSTEM_LIBUNWIND) || UNW_VERSION_MAJOR > 1 ||   \
-       (UNW_VERSION_MAJOR == 1 && UNW_VERSION_MINOR != 0 && UNW_VERSION_MINOR != 1))
 // Enable our memory manager only for libunwind with our patch or
 // on a newer release
 #    define JL_UNW_HAS_FORMAT_IP 1
-#  endif
 #else
 // Unwinding is disabled
 typedef int bt_context_t;
