Index: mayavi-4.5.0/docs/source/mayavi/sphinxext/sphinxext/compiler_unparse.py
===================================================================
--- mayavi-4.5.0.orig/docs/source/mayavi/sphinxext/sphinxext/compiler_unparse.py
+++ mayavi-4.5.0/docs/source/mayavi/sphinxext/sphinxext/compiler_unparse.py
@@ -190,12 +190,12 @@ class UnparseCompilerAst:
             if comma: self._write(", ")
             else: comma = True
             self._dispatch(e)
-        if t.starargs:
+        if getattr(t, 'starargs', None):
             if comma: self._write(", ")
             else: comma = True
             self._write("*")
             self._dispatch(t.starargs)
-        if t.kwargs:
+        if getattr(t, 'kwargs', None):
             if comma: self._write(", ")
             else: comma = True
             self._write("**")
