Index: install/cupy_builder/cupy_setup_build.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/install/cupy_builder/cupy_setup_build.py b/install/cupy_builder/cupy_setup_build.py
--- a/install/cupy_builder/cupy_setup_build.py	(revision 06dbe052f87a688231d2de843679a6595547807f)
+++ b/install/cupy_builder/cupy_setup_build.py	(date 1749101458921)
@@ -331,6 +331,10 @@
         settings['define_macros'].append(('__HIP_PLATFORM_AMD__', '1'))
         # deprecated since ROCm 4.2.0
         settings['define_macros'].append(('__HIP_PLATFORM_HCC__', '1'))
+        # Fix for ROCm 6.3.0
+        settings['define_macros'].append(('THRUST_DEVICE_SYSTEM', 'THRUST_DEVICE_SYSTEM_HIP'))
+        # Fix for https://github.com/ROCm/rocThrust/commit/df75ae8520fad81f1dc9a4b3b66358137c34daa6
+        settings['define_macros'].append(('__HIP__', '1'))
     settings['define_macros'].append(('CUPY_CACHE_KEY', ctx.cupy_cache_key))

     try:
Index: install/cupy_builder/install_build.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/install/cupy_builder/install_build.py b/install/cupy_builder/install_build.py
--- a/install/cupy_builder/install_build.py	(revision 06dbe052f87a688231d2de843679a6595547807f)
+++ b/install/cupy_builder/install_build.py	(date 1749101458930)
@@ -168,7 +168,8 @@
         library_dirs.append(os.path.join(rocm_path, 'lib'))

     if use_hip:
-        extra_compile_args.append('-std=c++11')
+        # ROCm5.3 and above requires c++14
+        extra_compile_args.append('-std=c++14')

     if PLATFORM_WIN32:
         nvtx_path = _environment.get_nvtx_path()
