*** HIP-rocm-4.2.0/hip-config.cmake.in.bak	2021-05-13 09:58:42.409738663 +0200
--- HIP-rocm-4.2.0/hip-config.cmake.in	2021-05-13 09:59:03.402050161 +0200
***************
*** 123,129 ****
    if(NOT WIN32)
      find_dependency(AMDDeviceLibs)
    endif()
!   set(AMDGPU_TARGETS "gfx900;gfx906;gfx908" CACHE STRING "AMD GPU targets to compile for")
    set(GPU_TARGETS "${AMDGPU_TARGETS}" CACHE STRING "GPU targets to compile for")
  endif()
  
--- 123,129 ----
    if(NOT WIN32)
      find_dependency(AMDDeviceLibs)
    endif()
!   set(AMDGPU_TARGETS "gfx803;gfx900;gfx906;gfx908" CACHE STRING "AMD GPU targets to compile for")
    set(GPU_TARGETS "${AMDGPU_TARGETS}" CACHE STRING "GPU targets to compile for")
  endif()
  
