*** HIP-rocm-3.10.0/CMakeLists.txt.orig	2020-12-03 19:05:48.525675785 +0100
--- HIP-rocm-3.10.0/CMakeLists.txt	2020-12-03 19:07:02.571936127 +0100
***************
*** 42,97 ****
  list(GET VERSION_LIST 0 HIP_VERSION_MAJOR)
  list(GET VERSION_LIST 1 HIP_VERSION_MINOR)
  
! find_package(Git)
! 
! # FIXME: Two different version strings used.
! if(GIT_FOUND)
!   # get date information based on UTC
!   # use the last two digits of year + week number + day in the week as HIP_VERSION_GITDATE
!   # use the commit date, instead of build date
!   # add xargs to remove strange trailing newline character
!   execute_process(COMMAND ${GIT_EXECUTABLE} show -s --format=@%ct
!     COMMAND xargs
!     COMMAND date -f - --utc +%y%U%w
!     RESULT_VARIABLE git_result
!     OUTPUT_VARIABLE git_output
!     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
!     OUTPUT_STRIP_TRAILING_WHITESPACE)
!   if(git_result EQUAL 0)
!     set(HIP_VERSION_GITDATE ${git_output})
!   endif()
! 
!   # get commit short hash
!   execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
!     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
!     RESULT_VARIABLE git_result
!     OUTPUT_VARIABLE git_output
!     OUTPUT_STRIP_TRAILING_WHITESPACE)
!   if(git_result EQUAL 0)
!     set(HIP_VERSION_GITHASH ${git_output})
!   endif()
! 
!   # get commit count
!   execute_process(COMMAND ${GIT_EXECUTABLE} rev-list --count HEAD
!     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
!     RESULT_VARIABLE git_result
!     OUTPUT_VARIABLE git_output
!     OUTPUT_STRIP_TRAILING_WHITESPACE)
!   if(git_result EQUAL 0)
!     set(HIP_VERSION_GITCOUNT ${git_output})
!   endif()
! 
!   set(HIP_VERSION_PATCH ${HIP_VERSION_GITDATE}-${HIP_VERSION_GITHASH})
! 
!   if(DEFINED ENV{ROCM_LIBPATCH_VERSION})
!     set(HIP_PACKAGING_VERSION_PATCH ${HIP_VERSION_GITDATE}.${HIP_VERSION_GITCOUNT}.$ENV{ROCM_LIBPATCH_VERSION})
!   else()
!     set(HIP_PACKAGING_VERSION_PATCH ${HIP_VERSION_GITDATE}.${HIP_VERSION_GITCOUNT}-${HIP_VERSION_GITHASH})
!   endif()
! else()
!   # FIXME: Some parts depend on this being set.
!   set(HIP_PACKAGING_VERSION_PATCH "0")
! endif()
  
  ## Debian package specific variables
  if ( DEFINED ENV{CPACK_DEBIAN_PACKAGE_RELEASE} )
--- 42,48 ----
  list(GET VERSION_LIST 0 HIP_VERSION_MAJOR)
  list(GET VERSION_LIST 1 HIP_VERSION_MINOR)
  
! set(HIP_PACKAGING_VERSION_PATCH "0")
  
  ## Debian package specific variables
  if ( DEFINED ENV{CPACK_DEBIAN_PACKAGE_RELEASE} )
