diff --git a/src/Makefile b/src/Makefile
index a99e52686..9b2b5705e 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -139,7 +139,7 @@ ULAPISRCS := rtapi/$(RTPREFIX)_ulapi.c
 
 # Each item in INCLUDES is transformed into a -I directive later on
 # The top directory is always included
-INCLUDES := .
+INCLUDES := . /usr/include/tirpc/
 
 USERSRCS :=
 PROGRAMS :=
@@ -203,6 +203,8 @@ else
 LDFLAGS := -Wl,-rpath-link,../lib
 endif
 
+LDFLAGS += -ltirpc
+
 # Rules to make .o (object) files
 $(sort $(CUSEROBJS)) : objects/%.o: %.c
  $(ECHO) Compiling $<
