diff --git a/src/rgw/driver/rados/rgw_zone.cc b/src/rgw/driver/rados/rgw_zone.cc
index c71656838df..67997d00b4b 100644
--- a/src/rgw/driver/rados/rgw_zone.cc
+++ b/src/rgw/driver/rados/rgw_zone.cc
@@ -195,7 +195,8 @@ int RGWZoneGroup::add_zone(const DoutPrefixProvider *dpp,
     zone.sync_from.erase(rm);
   }
 
-  zone.supported_features.insert(enable_features.begin(),
+  zone.supported_features.insert(boost::container::ordered_unique_range,
+                                 enable_features.begin(),
                                  enable_features.end());
 
   for (const auto& feature : disable_features) {
diff --git a/src/rgw/rgw_zone.cc b/src/rgw/rgw_zone.cc
index 90ff7cef106..59c9f322c0b 100644
--- a/src/rgw/rgw_zone.cc
+++ b/src/rgw/rgw_zone.cc
@@ -1423,7 +1423,9 @@ int add_zone_to_group(const DoutPrefixProvider* dpp, RGWZoneGroup& zonegroup,
   }
 
   // add/remove supported features
-  zone.supported_features.insert(enable_features.begin(),
+  zone.supported_features.insert(boost::container::ordered_unique_range,
+                                 enable_features.begin(),
+
                                  enable_features.end());
 
   for (const auto& feature : disable_features) {
