diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b03f76bff80..c670d055854 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -296,7 +296,9 @@ include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/src/xxHash")
 include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/src/rapidjson/include")
 
 option(WITH_FMT_HEADER_ONLY "use header-only version of fmt library" OFF)
-find_package(fmt 6.0.0 QUIET)
+set(WITH_FMT_VERSION "7.0.0" CACHE
+  STRING "build with fmt version")
+find_package(fmt ${WITH_FMT_VERSION} EXACT QUIET)
 if(fmt_FOUND)
   include_directories(SYSTEM "${fmt_INCLUDE_DIR}")
 else()
