diff --git a/src/test/librados/CMakeLists.txt b/src/test/librados/CMakeLists.txt
index e9338bbd464..f869ca4e821 100644
--- a/src/test/librados/CMakeLists.txt
+++ b/src/test/librados/CMakeLists.txt
@@ -58,9 +58,9 @@ add_executable(ceph_test_rados_api_aio_pp
 target_link_libraries(ceph_test_rados_api_aio_pp
   librados ${UNITTEST_LIBS} radostest-cxx)
 
-add_executable(ceph_test_rados_api_asio asio.cc)
-target_link_libraries(ceph_test_rados_api_asio global
-  librados ${UNITTEST_LIBS} Boost::coroutine Boost::context)
+#add_executable(ceph_test_rados_api_asio asio.cc)
+#target_link_libraries(ceph_test_rados_api_asio global
+#  librados ${UNITTEST_LIBS} Boost::coroutine Boost::context)
 
 add_executable(ceph_test_rados_api_list
   list.cc
@@ -144,7 +144,7 @@ target_link_libraries(ceph_test_rados_api_snapshots_pp
 install(TARGETS
   ceph_test_rados_api_aio
   ceph_test_rados_api_aio_pp
-  ceph_test_rados_api_asio
+  #ceph_test_rados_api_asio
   ceph_test_rados_api_c_read_operations
   ceph_test_rados_api_c_write_operations
   ceph_test_rados_api_cmd
