diff --git a/src/test/cls_fifo/CMakeLists.txt b/src/test/cls_fifo/CMakeLists.txt
index 3abf7634a2f..0344ce72161 100644
--- a/src/test/cls_fifo/CMakeLists.txt
+++ b/src/test/cls_fifo/CMakeLists.txt
@@ -1,3 +1,4 @@
+#[[
 add_executable(ceph_test_cls_fifo
   test_cls_fifo.cc
   )
@@ -32,3 +33,4 @@ target_link_libraries(ceph_bench_cls_fifo
 install(TARGETS
   ceph_test_cls_fifo
   DESTINATION ${CMAKE_INSTALL_BINDIR})
+]]
