--- a/src/pybind/mgr/dashboard/run-frontend-unittests.sh	2020-01-08 19:36:54.000000000 +0100
+++ b/src/pybind/mgr/dashboard/run-frontend-unittests.sh	2020-01-10 15:18:39.646510326 +0100
@@ -8,7 +8,7 @@ if [ `uname` != "FreeBSD" ]; then
 fi
 
 # Build
-npm run build -- --prod --progress=false || failed=true
+#npm run build -- --prod --progress=false || failed=true
 
 # Unit Tests
 config='src/unit-test-configuration.ts'
@@ -25,22 +25,22 @@ if [ -e ${config}_old ]; then
 fi
 
 # Linting
-npm run lint --silent
-if [ $? -gt 0 ]; then
-  failed=true
-  echo -e "\nTry running 'npm run fix' to fix some linting errors. \
-Some errors might need a manual fix."
-fi
-
-# I18N
-npm run i18n:extract
-i18n_lint=`awk '/<source> |<source>$| <\/source>/,/<\/context-group>/ {printf "%-4s ", NR; print}' src/locale/messages.xlf`
-if [[ ! -z $i18n_lint ]]; then
-  echo -e "The following source translations in 'messages.xlf' need to be \
-fixed, please check the I18N suggestions in 'HACKING.rst':\n"
-  echo "${i18n_lint}"
-  failed=true
-fi
+# npm run lint --silent
+# if [ $? -gt 0 ]; then
+#   failed=true
+#   echo -e "\nTry running 'npm run fix' to fix some linting errors. \
+# Some errors might need a manual fix."
+# fi
+# 
+# # I18N
+# npm run i18n:extract
+# i18n_lint=`awk '/<source> |<source>$| <\/source>/,/<\/context-group>/ {printf "%-4s ", NR; print}' src/locale/messages.xlf`
+# if [[ ! -z $i18n_lint ]]; then
+#   echo -e "The following source translations in 'messages.xlf' need to be \
+# fixed, please check the I18N suggestions in 'HACKING.rst':\n"
+#   echo "${i18n_lint}"
+#   failed=true
+# fi
 
 if [ `uname` != "FreeBSD" ]; then
   deactivate
