diff --git a/qa/tox.ini b/qa/tox.ini
index eac9e53701d..c276df199d3 100644
--- a/qa/tox.ini
+++ b/qa/tox.ini
@@ -15,7 +15,9 @@ commands = mypy {posargs:.}
 
 [testenv:import-tasks]
 basepython = python3
-deps = {env:TEUTHOLOGY_GIT:git+https://github.com/ceph/teuthology.git@master}#egg=teuthology[coverage,orchestra,test]
+deps = 
+  {env:TEUTHOLOGY_GIT:git+https://github.com/ceph/teuthology.git@master}#egg=teuthology[coverage,orchestra,test]
+  nose
 commands = python test_import.py {posargs:tasks/**/*.py}
 
 [testenv:pytest]
@@ -23,4 +25,4 @@ basepython = python3
 deps =
   {env:TEUTHOLOGY_GIT:git+https://github.com/ceph/teuthology.git@master}#egg=teuthology[test]
   httplib2
-commands = pytest -vv tasks/tests
\ No newline at end of file
+commands = pytest -vv tasks/tests
