diff --git a/pyproject.toml b/pyproject.toml
index 4728e80..4443288 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -9,7 +9,7 @@ requires = [
 build-backend = "setuptools.build_meta"
 
 [project]
-name = "bcrypt"
+name = "ceph_bcrypt"
 # When updating this, also update lib.rs
 version = "4.2.1"
 authors = [
@@ -41,7 +41,7 @@ homepage = "https://github.com/pyca/bcrypt/"
 [tool.setuptools]
 zip-safe = false
 package-dir = {"" = "src"}
-packages = ["bcrypt"]
+packages = ["ceph_bcrypt"]
 
 [tool.setuptools.dynamic]
 readme = {file = "README.rst", content-type = "text/x-rst"}
@@ -51,7 +51,7 @@ tests = ["pytest>=3.2.1,!=3.3.0"]
 typecheck = ["mypy"]

 [[tool.setuptools-rust.ext-modules]]
-target = "bcrypt._bcrypt"
+target = "ceph_bcrypt._bcrypt"
 path = "src/_bcrypt/Cargo.toml"
 py-limited-api = "auto"
 rust-version = ">=1.64.0"
diff --git a/tests/test_bcrypt.py b/tests/test_bcrypt.py
index 68c00fb..0661573 100644
--- a/tests/test_bcrypt.py
+++ b/tests/test_bcrypt.py
@@ -1,6 +1,6 @@
 import pytest
 
-import bcrypt
+import ceph_bcrypt as bcrypt
 
 _test_vectors = [
     (
