diff --git a/pyproject.toml b/pyproject.toml
index e365c8c..6e27a0d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -10,7 +10,7 @@ requires = [
 build-backend = "setuptools.build_meta"
 
 [project]
-name = "bcrypt"
+name = "ceph_bcrypt"
 # When updating this, also update lib.rs
 version = "4.1.2"
 authors = [
@@ -41,7 +41,7 @@ homepage = "https://github.com/pyca/bcrypt/"
 [tool.setuptools]
 zip-safe = false
 package-dir = {"" = "src"}
-packages = ["bcrypt"]
+packages = ["ceph_bcrypt"]
 
 [tool.setuptools.dynamic]
 readme = {file = "README.rst", content-type = "text/x-rst"}
@@ -57,7 +57,7 @@ select = ['E', 'F', 'I', 'N', 'W', 'UP', 'RUF']
 line-length = 79
 
 [tool.ruff.isort]
-known-first-party = ["bcrypt", "tests"]
+known-first-party = ["ceph_bcrypt", "tests"]
 
 [tool.mypy]
 show_error_codes = true
diff --git a/setup.py b/setup.py
index 13694c4..160abdd 100644
--- a/setup.py
+++ b/setup.py
@@ -35,7 +35,7 @@ try:
     setup(
         rust_extensions=[
             RustExtension(
-                "bcrypt._bcrypt",
+                "ceph_bcrypt._bcrypt",
                 "src/_bcrypt/Cargo.toml",
                 py_limited_api="auto",
                 rust_version=(
diff --git a/tests/test_bcrypt.py b/tests/test_bcrypt.py
index 68c00fb..0661573 100644
--- a/tests/test_bcrypt.py
+++ b/tests/test_bcrypt.py
@@ -1,6 +1,6 @@
 import pytest
 
-import bcrypt
+import ceph_bcrypt as bcrypt
 
 _test_vectors = [
     (
