diff --git a/Rcpp/inst/include/Rcpp/config.h b/Rcpp/inst/include/Rcpp/config.h
index fa75d9d..804089c 100644
--- a/Rcpp/inst/include/Rcpp/config.h
+++ b/Rcpp/inst/include/Rcpp/config.h
@@ -26,7 +26,7 @@
 #define RcppDevVersion(maj, min, rev, dev)  (((maj)*1000000) + ((min)*10000) + ((rev)*100) + (dev))
 
 // the currently released version
-#define RCPP_VERSION            Rcpp_Version(1,0,11)
+#define RCPP_VERSION            Rcpp_Version(1,0,12)
 #define RCPP_VERSION_STRING     "1.0.12"
 
 // the current source snapshot (using four components, if a fifth is used in DESCRIPTION we ignore it)
