--- qgroundcontrol-4.0.0/QGCSetup.pri	2020-02-26 08:29:42.000000000 +1000
+++ qgroundcontrol-4.0.0/QGCSetup.pri.new	2020-02-26 12:56:25.163049000 +1000
@@ -124,9 +124,9 @@
     !contains(DEFINES, __rasp_pi2__) {
         # Some Qt distributions link with *.so.56
         QT_LIB_LIST += \
-            libicudata.so.56 \
-            libicui18n.so.56 \
-            libicuuc.so.56
+            libicudata.so.* \
+            libicui18n.so.* \
+            libicuuc.so.*
     }
 
     for(QT_LIB, QT_LIB_LIST) {
