diff --color --unified --recursive --text orig/pyproject.toml new/pyproject.toml
--- orig/pyproject.toml	2022-04-04 20:14:29.949020090 +0200
+++ new/pyproject.toml	2022-04-04 21:24:32.652332853 +0200
@@ -2,12 +2,8 @@
 requires = [
   "setuptools",
   "wheel",
-  "conan>=1.40.0",
   "scikit-build>=0.11.0",
-  "cmake!=3.17.1,!=3.17.0",
-  "ninja",
   "pybind11>2.6",
-  "oldest-supported-numpy; python_version>'3.7' or platform_machine=='aarch64' or platform_python_implementation=='PyPy'",
   "numpy==1.16.3; python_version<='3.7' and platform_machine!='aarch64' or platform_python_implementation=='PyPy'",
 ]
 build-backend = "setuptools.build_meta"
diff --color --unified --recursive --text orig/setup.py new/setup.py
--- orig/setup.py	2022-04-04 20:14:29.949020090 +0200
+++ new/setup.py	2022-04-04 21:15:05.645526211 +0200
@@ -72,7 +72,6 @@
 
 setup_requirements = common_requirements + [
     'scikit-build>=0.11.0',
-    'cmake!=3.17,!=3.17.0',
     'pybind11>=2.6',
 ]
 
