diff --color --color -uraBN gmsh-4.13.1-source/contrib/hxt/tetBR/src/hxt_boundary_recovery.cxx gmsh-4.13.1-source-patched/contrib/hxt/tetBR/src/hxt_boundary_recovery.cxx
--- gmsh-4.13.1-source/contrib/hxt/tetBR/src/hxt_boundary_recovery.cxx	2023-07-21 13:57:22.000000000 -0500
+++ gmsh-4.13.1-source-patched/contrib/hxt/tetBR/src/hxt_boundary_recovery.cxx	2025-06-30 13:32:04.443169381 -0500
@@ -1,7 +1,7 @@
+#include "hxt_omp.h"
 extern "C" {
 #include "hxt_mesh.h"
 #include "hxt_tools.h"
-#include "hxt_omp.h"
 #include "predicates.h"
 }
 
diff --color --color -uraBN gmsh-4.13.1-source/src/mesh/meshGRegionHxt.cpp gmsh-4.13.1-source-patched/src/mesh/meshGRegionHxt.cpp
--- gmsh-4.13.1-source/src/mesh/meshGRegionHxt.cpp	2024-03-29 17:07:01.000000000 -0500
+++ gmsh-4.13.1-source-patched/src/mesh/meshGRegionHxt.cpp	2025-06-30 13:32:57.224171994 -0500
@@ -24,8 +24,8 @@
 
 #if defined(HAVE_HXT)
 
-extern "C" {
 #include "hxt_omp.h"
+extern "C" {
 #include "hxt_tetMesh.h"
 #include "hxt_tetDelaunay.h"
 }
