diff --color -urN flang-rocm-5.0.1-unpatched/runtime/flang/alarm3f.c flang-rocm-5.0.1/runtime/flang/alarm3f.c
--- flang-rocm-5.0.1-unpatched/runtime/flang/alarm3f.c  2022-03-01 13:57:36.463305752 +0100
+++ flang-rocm-5.0.1/runtime/flang/alarm3f.c    2022-03-01 13:59:55.300264387 +0100
@@ -16,7 +16,7 @@
 /*
 extern void (*signal(int, void (*)(int)))(int);
 */
-extern int alarm();
+extern unsigned int alarm(unsigned int __seconds);

 int ENT3F(ALARM, alarm)(int *time, void (*proc)())
 {
