diff --color -urN aomp-rocm-4.3.1/bin/build_openmp.sh aomp-rocm-4.3.1-patched/bin/build_openmp.sh
--- aomp-rocm-4.3.1/bin/build_openmp.sh	2021-08-27 05:56:05.000000000 +0200
+++ aomp-rocm-4.3.1-patched/bin/build_openmp.sh	2021-10-29 19:43:56.338668126 +0200
@@ -99,21 +99,11 @@
 
 GCCLOC=$(getgcc8orless)
 GXXLOC=$(getgxx8orless)
-if [ "$GCCLOC" == "" ] ; then
-   echo "ERROR: NO ADEQUATE gcc"
-   echo "       Please install gcc-5, gcc-7, or gcc-8"
-   exit 1
-fi
-if [ "$GXXLOC" == "" ] ; then
-   echo "ERROR: NO ADEQUATE g++"
-   echo "       Please install g++-5, g++-7, or g++-8"
-   exit 1
-fi
 
 GFXSEMICOLONS=`echo $GFXLIST | tr ' ' ';' `
 #COMMON_CMAKE_OPTS="#-DOPENMP_TEST_C_COMPILER=$AOMP/bin/clang
 #-DOPENMP_TEST_CXX_COMPILER=$AOMP/bin/clang++
 
 COMMON_CMAKE_OPTS="-DDEVICELIBS_ROOT=$DEVICELIBS_ROOT
 -DOPENMP_ENABLE_LIBOMPTARGET=1
 -DOPENMP_ENABLE_LIBOMPTARGET_HSA=1
@@ -127,6 +119,16 @@
 -DCMAKE_PREFIX_PATH=$ROCM_DIR;$ROCM_DIR/include/hsa"
 
 if [ "$AOMP_BUILD_CUDA" == 1 ] ; then
+  if [ "$GCCLOC" == "" ] ; then
+     echo "ERROR: NO ADEQUATE gcc"
+     echo "       Please install gcc-5, gcc-7, or gcc-8"
+     exit 1
+  fi
+  if [ "$GXXLOC" == "" ] ; then
+     echo "ERROR: NO ADEQUATE g++"
+     echo "       Please install g++-5, g++-7, or g++-8"
+     exit 1
+  fi
    COMMON_CMAKE_OPTS="$COMMON_CMAKE_OPTS
 -DLIBOMPTARGET_NVPTX_ENABLE_BCLIB=ON
 -DLIBOMPTARGET_NVPTX_CUDA_COMPILER=$AOMP/bin/clang++
