diff --color -urN aomp-rocm-4.5.0-unpatched/bin/build_flang_runtime.sh aomp-rocm-4.5.0/bin/build_flang_runtime.sh
--- aomp-rocm-4.5.0-unpatched/bin/build_flang_runtime.sh	2021-11-16 16:50:07.010969269 +0100
+++ aomp-rocm-4.5.0/bin/build_flang_runtime.sh	2021-11-16 16:57:38.658874349 +0100
@@ -59,10 +59,10 @@
 MYCMAKEOPTS="-DCMAKE_BUILD_TYPE=$BUILD_TYPE \
 -DCMAKE_INSTALL_PREFIX=$INSTALL_FLANG \
 -DLLVM_ENABLE_ASSERTIONS=ON \
--DLLVM_CONFIG=$OUT_DIR/llvm/bin/llvm-config \
--DCMAKE_CXX_COMPILER=$OUT_DIR/llvm/bin/clang++ \
--DCMAKE_C_COMPILER=$OUT_DIR/llvm/bin/clang \
--DCMAKE_Fortran_COMPILER=$OUT_DIR/llvm/bin/flang \
+-DLLVM_CONFIG=$ROCM_DIR/llvm/bin/llvm-config \
+-DCMAKE_CXX_COMPILER=$ROCM_DIR/llvm/bin/clang++ \
+-DCMAKE_C_COMPILER=$ROCM_DIR/llvm/bin/clang \
+-DCMAKE_Fortran_COMPILER=$ROCM_DIR/llvm/bin/flang \
 -DLLVM_TARGETS_TO_BUILD=$TARGETS_TO_BUILD \
 -DLLVM_INSTALL_RUNTIME=ON \
 -DFLANG_BUILD_RUNTIME=ON \
diff --color -urN aomp-rocm-4.5.0-unpatched/bin/build_flang.sh aomp-rocm-4.5.0/bin/build_flang.sh
--- aomp-rocm-4.5.0-unpatched/bin/build_flang.sh	2021-11-16 16:50:07.010969269 +0100
+++ aomp-rocm-4.5.0/bin/build_flang.sh	2021-11-16 16:55:45.538909397 +0100
@@ -50,9 +50,9 @@
 MYCMAKEOPTS="-DCMAKE_BUILD_TYPE=$BUILD_TYPE \
 -DCMAKE_INSTALL_PREFIX=$INSTALL_FLANG \
 -DLLVM_ENABLE_ASSERTIONS=ON \
--DLLVM_CONFIG=$OUT_DIR/llvm/bin/llvm-config \
--DCMAKE_CXX_COMPILER=$OUT_DIR/llvm/bin/clang++ \
--DCMAKE_C_COMPILER=$OUT_DIR/llvm/bin/clang \
+-DLLVM_CONFIG=$ROCM_DIR/llvm/bin/llvm-config \
+-DCMAKE_CXX_COMPILER=$ROCM_DIR/llvm/bin/clang++ \
+-DCMAKE_C_COMPILER=$ROCM_DIR/llvm/bin/clang \
 -DCMAKE_Fortran_COMPILER=gfortran \
 -DLLVM_TARGETS_TO_BUILD=$TARGETS_TO_BUILD \
 -DCMAKE_C_FLAGS=-I$COMP_INC_DIR \
diff --color -urN aomp-rocm-4.5.0-unpatched/bin/build_pgmath.sh aomp-rocm-4.5.0/bin/build_pgmath.sh
--- aomp-rocm-4.5.0-unpatched/bin/build_pgmath.sh	2021-11-16 18:52:19.761286900 +0100
+++ aomp-rocm-4.5.0/bin/build_pgmath.sh	2021-11-16 18:53:21.787423127 +0100
@@ -41,7 +41,7 @@
    fi
 fi
 
-COMP_INC_DIR=$(ls -d $OUT_DIR/llvm/lib/clang/*/include )
+COMP_INC_DIR=$(ls -d $ROCM_DIR/llvm/lib/clang/*/include )
 
 if [ "$AOMP_PROC" == "ppc64le" ] ; then
     MYCMAKEOPTS="-DCMAKE_BUILD_TYPE=$BUILD_TYPE -DCMAKE_INSTALL_PREFIX=$INSTALL_FLANG -DLLVM_ENABLE_ASSERTIONS=ON $AOMP_ORIGIN_RPATH -DCMAKE_Fortran_COMPILER=$AOMP_INSTALL_DIR/bin/flang -DLLVM_TARGETS_TO_BUILD=$TARGETS_TO_BUILD -DCMAKE_C_FLAGS=-I$COMP_INC_DIR -DCMAKE_CXX_FLAGS=-I$COMP_INC_DIR"
@@ -51,8 +51,8 @@
     -DCMAKE_INSTALL_PREFIX=$INSTALL_FLANG \
     -DLLVM_ENABLE_ASSERTIONS=ON \
     -DLLVM_CONFIG=$INSTALL_FLANG/bin/llvm-config \
-    -DCMAKE_CXX_COMPILER=$OUT_DIR/llvm/bin/clang++ \
-    -DCMAKE_C_COMPILER=$OUT_DIR/llvm/bin/clang \
+    -DCMAKE_CXX_COMPILER=$ROCM_DIR/llvm/bin/clang++ \
+    -DCMAKE_C_COMPILER=$ROCM_DIR/llvm/bin/clang \
     -DLLVM_TARGETS_TO_BUILD=$TARGETS_TO_BUILD \
     -DCMAKE_C_FLAGS=-I$COMP_INC_DIR \
     -DCMAKE_CXX_FLAGS=-I$COMP_INC_DIR \
