diff --color -urN llvm-project-rocm-5.0.1-unpatched/openmp/libomptarget/hostrpc/src/hostrpc.h llvm-project-rocm-5.0.1/openmp/libomptarget/hostrpc/src/hostrpc.h
--- llvm-project-rocm-5.0.1-unpatched/openmp/libomptarget/hostrpc/src/hostrpc.h 2022-03-01 13:37:53.355306931 +0100
+++ llvm-project-rocm-5.0.1/openmp/libomptarget/hostrpc/src/hostrpc.h   2022-03-01 13:54:46.401334335 +0100
@@ -41,6 +41,14 @@
 #include <stdint.h>
 #include <stdio.h>

+#if defined(fprintf)
+    #undef fprintf
+#endif
+
+#if defined(printf)
+    #undef printf
+#endif
+
 //  These are the interfaces for the device stubs */
 EXTERN int fprintf(FILE *, const char *, ...);
 EXTERN char *fprintf_allocate(uint32_t bufsz);
