--- gtsam-4.0.2/CMakeLists.txt.old	2019-10-12 17:14:32.067880953 -0700
+++ gtsam-4.0.2/CMakeLists.txt	2019-10-12 17:14:43.756628849 -0700
@@ -185,12 +185,6 @@
   optimized ${Boost_THREAD_LIBRARY_RELEASE}
   optimized ${Boost_DATE_TIME_LIBRARY_RELEASE}
   optimized ${Boost_REGEX_LIBRARY_RELEASE}
-  debug ${Boost_SERIALIZATION_LIBRARY_DEBUG}
-  debug ${Boost_SYSTEM_LIBRARY_DEBUG}
-  debug ${Boost_FILESYSTEM_LIBRARY_DEBUG}
-  debug ${Boost_THREAD_LIBRARY_DEBUG}
-  debug ${Boost_DATE_TIME_LIBRARY_DEBUG}
-  debug ${Boost_REGEX_LIBRARY_DEBUG}
 )
 message(STATUS "GTSAM_BOOST_LIBRARIES: ${GTSAM_BOOST_LIBRARIES}")
 if (GTSAM_DISABLE_NEW_TIMERS)
@@ -201,8 +195,6 @@
       list(APPEND GTSAM_BOOST_LIBRARIES
         optimized ${Boost_TIMER_LIBRARY_RELEASE}
         optimized ${Boost_CHRONO_LIBRARY_RELEASE}
-        debug ${Boost_TIMER_LIBRARY_DEBUG}
-        debug ${Boost_CHRONO_LIBRARY_DEBUG}
         )
     else()
       list(APPEND GTSAM_BOOST_LIBRARIES rt) # When using the header-only boost timer library, need -lrt
