diff --git a/src/Makevars b/src/Makevars
index 30c0d3a..bb2b1e4 100644
--- a/src/Makevars
+++ b/src/Makevars
@@ -1,5 +1,5 @@
 # BOOST_NO_AUTO_PTR: https://github.com/eddelbuettel/bh/issues/57#issuecomment-406261595
-PKG_CPPFLAGS=-Ivendor \
+PKG_CPPFLAGS=-I. \
              -DRSQLITE_USE_BUNDLED_SQLITE \
              -DSQLITE_ENABLE_RTREE \
              -DSQLITE_ENABLE_FTS3 \
@@ -15,7 +15,7 @@ PKG_CPPFLAGS=-Ivendor \
 PKG_CXXFLAGS=$(CXX_VISIBILITY)
 PKG_CFLAGS=$(C_VISIBILITY)
 
-PKG_LIBS = vendor/sqlite3/sqlite3.o
+PKG_LIBS = -lsqlite3
 
 .PHONY: all
 
diff --git a/src/import-file.c b/src/import-file.c
index c2bb19f..6dc4311 100644
--- a/src/import-file.c
+++ b/src/import-file.c
@@ -18,7 +18,7 @@
 
 #include <R.h>
 #include <Rinternals.h>
-#include "vendor/sqlite3/sqlite3.h"
+#include <sqlite3.h>
 
 char* RS_sqlite_getline(FILE* in, const char* eol);
 
