diff --git a/applications/utilities/surface/surfaceBooleanFeatures/surfaceBooleanFeatures.C b/applications/utilities/surface/surfaceBooleanFeatures/surfaceBooleanFeatures.C
index 9bd01d4e..a920b754 100644
--- a/applications/utilities/surface/surfaceBooleanFeatures/surfaceBooleanFeatures.C
+++ b/applications/utilities/surface/surfaceBooleanFeatures/surfaceBooleanFeatures.C
@@ -647,7 +647,7 @@ labelPair edgeIntersectionsCGAL
             // Get intersection object
             if
             (
-                const Point* ptPtr = boost::get<Point>(&(intersect->first))
+                const Point* ptPtr = std::get_if<Point>(&(intersect->first))
             )
             {
                 point pt
@@ -679,7 +679,7 @@ labelPair edgeIntersectionsCGAL
             }
             else if
             (
-                const Segment* sPtr = boost::get<Segment>(&(intersect->first))
+                const Segment* sPtr = std::get_if<Segment>(&(intersect->first))
             )
             {
                 #if defined (CGAL_VERSION_NR) && (CGAL_VERSION_NR < 1041400000)
