diff --git a/StanHeaders/R/Flags.R b/StanHeaders/R/Flags.R
index 244a856..d74217d 100644
--- a/StanHeaders/R/Flags.R
+++ b/StanHeaders/R/Flags.R
@@ -5,8 +5,12 @@ CxxFlags <- function(as_character = FALSE) {
     TBB_INC <- system.file("include", package = "RcppParallel", mustWork = TRUE)
   }
 
-  if (file.exists(file.path(TBB_INC, "tbb", "version.h"))) {
-    CXXFLAGS <- paste0("-I", shQuote(TBB_INC), " -D_REENTRANT -DSTAN_THREADS -DTBB_INTERFACE_NEW")
+  TBB_DIR <- file.path(TBB_INC, "tbb")
+  if (!dir.exists(TBB_DIR)) {
+      TBB_DIR <- "/usr/include/tbb"
+  }
+
+  if (file.exists(file.path(TBB_DIR, "version.h"))) {
   } else {
     CXXFLAGS <- paste0("-I", shQuote(TBB_INC), " -D_REENTRANT -DSTAN_THREADS")
     # RcppParallel TBB does not have assembly code for Windows ARM64
@@ -30,7 +34,7 @@ LdFlags <- function(as_character = FALSE) {
   if (dir.exists(TBB_LIB)) {
     TBB_LIB <- normalizePath(TBB_LIB)
   } else {
-    TBB_LIB <- system.file("lib", .Platform$r_arch, package = "RcppParallel", mustWork = TRUE)
+    TBB_LIB <- file.path(system.file(package = "RcppParallel", mustWork = TRUE), "lib")
   }
 
   PKG_LIBS <- paste0("-L", shQuote(TBB_LIB))
@@ -38,7 +42,12 @@ LdFlags <- function(as_character = FALSE) {
   if (!(.Platform$OS.type == "windows" && R.version$arch == "aarch64")) {
     PKG_LIBS <- paste0(PKG_LIBS, " -Wl,-rpath,", shQuote(TBB_LIB))
   }
-  PKG_LIBS <- paste0(PKG_LIBS, " -ltbb -ltbbmalloc")
+  if (dir.exists(TBB_LIB)) {
+    TBB_LIB <- normalizePath(TBB_LIB)
+    PKG_LIBS <- paste0("-L", shQuote(TBB_LIB), " -Wl,-rpath,", shQuote(TBB_LIB), " -ltbb -ltbbmalloc")
+  } else {
+    PKG_LIBS <- "-ltbb -ltbbmalloc"
+  }
 
   if (isTRUE(as_character)) return(PKG_LIBS)
   cat(PKG_LIBS, " ")
