--- a/py/selenium/webdriver/ie/options.py
+++ b/py/selenium/webdriver/ie/options.py
@@ -59,8 +59,8 @@ def __set__(self, obj, value) -> None:
             raise ValueError(f"{self.name} should be of type {self.expected_type.__name__}")
 
         if self.name == "elementScrollBehavior" and value not in [
-            ElementScrollBehavior.TOP.value,
-            ElementScrollBehavior.BOTTOM.value,
+            ElementScrollBehavior.TOP,
+            ElementScrollBehavior.BOTTOM,
         ]:
             raise ValueError("Element Scroll Behavior out of range.")
         obj._options[self.name] = value
@@ -105,7 +105,7 @@ class Options(ArgOptions):
     `value`: `int` (Timeout) in milliseconds
     """

-    element_scroll_behavior = _IeOptionsDescriptor(ELEMENT_SCROLL_BEHAVIOR, int)
+    element_scroll_behavior = _IeOptionsDescriptor(ELEMENT_SCROLL_BEHAVIOR, Enum)
     """Gets and Sets `element_scroll_behavior`

     Usage
