diff --unified --recursive --text --color Trilinos-trilinos-release-13-0-1.old/packages/PyTrilinos/src/gen_teuchos_rcp.py.in Trilinos-trilinos-release-13-0-1.new/packages/PyTrilinos/src/gen_teuchos_rcp.py.in
--- Trilinos-trilinos-release-13-0-1.old/packages/PyTrilinos/src/gen_teuchos_rcp.py.in  2020-11-14 01:13:13.000000000 +0100
+++ Trilinos-trilinos-release-13-0-1.new/packages/PyTrilinos/src/gen_teuchos_rcp.py.in  2020-12-21 09:48:13.021304841 +0100
@@ -53,7 +53,7 @@
 ################################################################################

 def get_mpi_version():
-    header = "${MPI_BASE_DIR}/include/mpi.h"
+    header = os.path.join(MPI_BASE_DIR, "include/mpi.h")
     version = ""
     for line in open(header, 'r').readlines():
         if "MPI_VERSION" in line:
