diff -aur speech_tools.old//config/compilers/jdk.mak speech_tools/config/compilers/jdk.mak
--- speech_tools.old//config/compilers/jdk.mak	2001-04-04 19:55:32.000000000 +0800
+++ speech_tools/config/compilers/jdk.mak	2010-12-08 12:14:56.076668109 +0800
@@ -52,7 +52,7 @@
 endif
 endif
 
-JAVAFLAGS = -depend
+JAVAFLAGS =
 
 DEBUG_JAVAFLAGS = -g
 
diff -aur speech_tools.old//config/config.in speech_tools/config/config.in
--- speech_tools.old//config/config.in	2008-10-29 03:20:49.000000000 +0800
+++ speech_tools/config/config.in	2010-12-08 12:17:50.110001443 +0800
@@ -15,7 +15,7 @@
 ## You may need to set this explicitly if automounter or NFS
 ## side effects cause problems
 
-EST_HOME := $(shell (cd $(EST); pwd))
+EST_HOME := /usr
 
 ###########################################################################
 ## System type. 
@@ -28,14 +28,14 @@
 ## 
 ## Examples: sparc_SunOS5 intel_Linux2.0
 
-SYSTEM_TYPE=$(MACHINETYPE)_$(OSTYPE)$(OSREV)
+SYSTEM_TYPE=Linux
 
 ###########################################################################
 ## Compiler.
 ## The definitions are in compilers/$(COMPILER).mak
 ## Examples: gcc suncc egcs gcc28
 
-COMPILER=@COMPILERTYPE@
+COMPILER=gcc
 
 ###########################################################################
 ## Java system to use if you include the Java interface.
@@ -61,7 +61,7 @@
 # VERBOSE=1
 #DEBUG=1
 # PROFILE=gprof
-#SHARED=2
+SHARED=2
 
 ## Directory specific selections which override the above
 
@@ -98,7 +98,7 @@
 ## It may not work under all systems, so may be optionally omitted.
 INCLUDE_MODULES += EDITLINE
 
-TERMCAPLIB = @TERMCAPLIB@
+TERMCAPLIB = 
 # speech recognition
 #INCLUDE_MODULES += ASR
 
@@ -175,6 +175,6 @@
 CONFIG_WRAPPER_LANGUAGES = PYTHON
 
 # Language specific includes should be set to correct site paths
-CONFIG_PYTHON_INCLUDES= -I/usr/include/python2.2/
+CONFIG_PYTHON_INCLUDES= -I/usr/include/python2.7/
 ##
 ###################################################################
Only in speech_tools/config: config.in.orig
diff -aur speech_tools.old//config/systems/default.mak speech_tools/config/systems/default.mak
--- speech_tools.old//config/systems/default.mak	2005-07-17 21:06:49.000000000 +0800
+++ speech_tools/config/systems/default.mak	2010-12-08 12:14:56.076668109 +0800
@@ -40,7 +40,7 @@
 ###########################################################################
 ## Installation directories
 
-INSTALL_PREFIX=/usr/local
+INSTALL_PREFIX=/usr
 
 BINDIR=$(INSTALL_PREFIX)/bin
 LIBDIR=$(INSTALL_PREFIX)/lib
@@ -63,8 +63,8 @@
 ###########################################################################
 ## Where to find Enlightenment Speech Demon
 
-ESD_INCLUDE = /usr/local/include
-ESD_LIB = /usr/local/lib
+ESD_INCLUDE = /usr/include
+ESD_LIB = /usr/lib
 
 ###########################################################################
 ## Where to find X11
@@ -75,14 +75,14 @@
 ###########################################################################
 ## TCL support
 
-TCL_INCLUDE = /usr/local/include
-TCL_LIB     = /usr/local/lib
-TCL_LIBRARY = -ltcl7.6
+TCL_INCLUDE = /usr/include
+TCL_LIB     = /usr/lib
+TCL_LIBRARY = -ltcl
 
 ###########################################################################
 ## Efence library for malloc debugging
 
-EFENCE_LIB = /usr/local/lib
+EFENCE_LIB = /usr/lib
 
 ###########################################################################
 ## Commands.
diff -aur speech_tools.old//config/systems/Linux.mak speech_tools/config/systems/Linux.mak
--- speech_tools.old//config/systems/Linux.mak	2001-04-04 19:55:32.000000000 +0800
+++ speech_tools/config/systems/Linux.mak	2010-12-08 12:14:56.076668109 +0800
@@ -40,13 +40,13 @@
 
 include $(EST)/config/systems/default.mak
 
-DEFAULT_JAVA_HOME=/usr/lib/jdk-1.1.6
+DEFAULT_JAVA_HOME=/usr/lib/jvm/java-6-openjdk
 JAVA=/usr/bin/java
 JAVAC=/usr/bin/javac
 JAVAH=/usr/bin/javah
 
 TCL_LIBRARY = -ltcl
-OS_LIBS = -ldl 
+OS_LIBS = -ldl -lncurses 
 
 ## the native audio module for this type of system
 NATIVE_AUDIO_MODULE = LINUX16
