diff --git a/torch/utils/cpp_extension.py b/torch/utils/cpp_extension.py
index ca673033e1..c79ce8d37b 100644
--- a/torch/utils/cpp_extension.py
+++ b/torch/utils/cpp_extension.py
@@ -1760,7 +1760,7 @@ def _write_ninja_file_to_build_library(path,
             common_cflags.append(f'-DPYBIND11_{pname}=\\"{pval}\\"')
 
     common_cflags += [f'-I{include}' for include in user_includes]
-    common_cflags += [f'-isystem {include}' for include in system_includes]
+    common_cflags += [f'-I{include}' for include in system_includes]
 
     common_cflags += ['-D_GLIBCXX_USE_CXX11_ABI=' + str(int(torch._C._GLIBCXX_USE_CXX11_ABI))]
 
