--- a/utils/cpp_extension.py	2019-12-29 12:59:36.083692770 -0300
+++ b/utils/cpp_extension.py	2020-01-08 09:52:16.435316701 -0300
@@ -1098,7 +1098,7 @@
     if BUILD_NAMEDTENSOR:
         common_cflags.append('-DBUILD_NAMEDTENSOR')
     common_cflags += ['-I{}'.format(include) for include in user_includes]
-    common_cflags += ['-isystem {}'.format(include) for include in system_includes]
+    common_cflags += ['-I{}'.format(include) for include in system_includes]
 
     common_cflags += ['-D_GLIBCXX_USE_CXX11_ABI=' + str(int(torch._C._GLIBCXX_USE_CXX11_ABI))]

