--- qgroundcontrol-3.5.5/QGCCommon.pri.old	2019-10-21 11:49:43.026322877 +1000
+++ qgroundcontrol-3.5.5/QGCCommon.pri	2019-10-21 11:49:33.886275250 +1000
@@ -26,6 +26,7 @@
             message("Linux clang")
             QMAKE_CXXFLAGS += -Qunused-arguments -fcolor-diagnostics
         }
+        QMAKE_CXXFLAGS += -Wno-address-of-packed-member
     } else : linux-rasp-pi2-g++ {
         message("Linux R-Pi2 build")
         CONFIG += LinuxBuild
