diff --unified --recursive --text qgroundcontrol-3.5.0/QGCSetup.pri qgroundcontrol-3.5.0.new/QGCSetup.pri
--- qgroundcontrol-3.5.0/QGCSetup.pri	2019-02-05 07:38:43.000000000 +1000
+++ qgroundcontrol-3.5.0.new/QGCSetup.pri	2019-04-18 18:35:13.792079826 +1000
@@ -135,10 +135,11 @@
         libQt5TextToSpeech.so.5
 
     !contains(DEFINES, __rasp_pi2__) {
+        # Some Qt distributions link with *.so.56
         QT_LIB_LIST += \
-            libicudata.so.56 \
-            libicui18n.so.56 \
-            libicuuc.so.56
+            libicudata.so.* \
+            libicui18n.so.* \
+            libicuuc.so.*
     }
 
     for(QT_LIB, QT_LIB_LIST) {
