diff --color --unified --recursive --text orig/src/main.qml new/src/main.qml
--- orig/src/main.qml	2023-11-18 09:52:10.767398847 +0100
+++ new/src/main.qml	2023-11-18 09:53:12.521334474 +0100
@@ -1500,13 +1500,6 @@
                     }
                 }
             }
-
-            if (imageWriter.getBoolSetting("check_version") && "latest_version" in imager && "url" in imager) {
-                if (!imageWriter.isEmbeddedMode() && imageWriter.isVersionNewer(imager["latest_version"])) {
-                    updatepopup.url = imager["url"]
-                    updatepopup.openPopup()
-                }
-            }
             if ("default_os" in imager) {
                 selectNamedOS(imager["default_os"], osmodel)
             }
