--- OpenBoard-1.5.3/OpenBoard.pro	2019-04-10 09:04:25.000000000 +0200
+++ OpenBoard-1.5.3-quazip/OpenBoard.pro	2019-04-12 08:54:40.265598779 +0200
@@ -71,9 +71,7 @@
 INCLUDEPATH += plugins/cffadaptor/src
 
 #ThirdParty
-DEPENDPATH += $$THIRD_PARTY_PATH/quazip/
-INCLUDEPATH += $$THIRD_PARTY_PATH/quazip/
-include($$THIRD_PARTY_PATH/quazip/quazip.pri)
+INCLUDEPATH += /usr/include/QuaZip-Qt5-1.0/quazip
 
 FORMS += resources/forms/mainWindow.ui \
    resources/forms/preferences.ui \
--- OpenBoard-1.5.3/plugins/cffadaptor/UBCFFAdaptor.pro	2019-04-10 09:04:25.000000000 +0200
+++ OpenBoard-1.5.3-quazip/plugins/cffadaptor/UBCFFAdaptor.pro	2019-04-12 08:53:45.622844493 +0200
@@ -39,7 +38,7 @@
     QMAKE_POST_LINK += bash postScript_mac.sh "$$DESTDIR" "$$SANKORE_DIR/$$BUILD_DIR/product/Open-Sankore.app/Contents/MacOS"
 }
 
-LIBS     += "-L$$THIRD_PARTY_PATH/quazip/lib/$$SUB_DIR" "-lquazip"
+LIBS     += "-lquazip1-qt5"
 
 QT       += xml xmlpatterns core
 QT       += gui
@@ -51,9 +50,7 @@
 
 INCLUDEPATH += src
 
-DEPENDPATH  += $$THIRD_PARTY_PATH/quazip/
-INCLUDEPATH += $$THIRD_PARTY_PATH/quazip/
-include($$THIRD_PARTY_PATH/quazip/quazip.pri)
+INCLUDEPATH += /usr/include/QuaZip-Qt5-1.0/quazip
 
 
 SOURCES += \
--- OpenBoard-1.5.4.orig/libs.pri	2020-10-11 14:17:54.587596163 +0200
+++ OpenBoard-1.5.4/libs.pri	2020-10-11 14:13:49.516394503 +0200
@@ -23,7 +23,7 @@
 }
 
 QUAZIP_DIR   = "/usr"
-LIBS        += "-lquazip5"
+LIBS        += "-lquazip1-qt5"
 
 LIBS        += "-lpoppler"
 INCLUDEPATH += "/usr/include/poppler"
