diff -Naur OpenBoard-1.6.0a3.orig/OpenBoard.pro OpenBoard-1.6.0a3/OpenBoard.pro
--- OpenBoard-1.6.0a3.orig/OpenBoard.pro	2020-05-22 18:40:49.000000000 +0200
+++ OpenBoard-1.6.0a3/OpenBoard.pro	2020-10-12 07:26:16.628748656 +0200
@@ -433,8 +431,8 @@
     LIBS += -lcrypto
     #LIBS += -lprofiler
     LIBS += -lX11
-    LIBS += -lquazip5
-    INCLUDEPATH += "/usr/include/quazip"
+    LIBS += "-lquazip1-qt5"
+    INCLUDEPATH += /usr/include/QuaZip-Qt5-1.1/quazip
 
     LIBS += -lpoppler
     INCLUDEPATH += "/usr/include/poppler"
diff -Naur OpenBoard-1.6.0a3.orig/plugins/cffadaptor/UBCFFAdaptor.pro OpenBoard-1.6.0a3/plugins/cffadaptor/UBCFFAdaptor.pro
--- OpenBoard-1.6.0a3.orig/plugins/cffadaptor/UBCFFAdaptor.pro	2020-05-22 18:40:49.000000000 +0200
+++ OpenBoard-1.6.0a3/plugins/cffadaptor/UBCFFAdaptor.pro	2020-10-12 07:20:08.496096917 +0200
@@ -13,9 +13,6 @@
 linux-g++-32: SUB_DIR = linux
 linux-g++-64: SUB_DIR = linux
 
-THIRD_PARTY_PATH = ../../../Sankore-ThirdParty
-QUAZIP_DIR   = "$$PWD/../../../Sankore-ThirdParty/quazip/quazip-0.3"
-
 BUILD_DIR = build/$$SUB_DIR
 CONFIG(debug, debug|release):BUILD_DIR = $$BUILD_DIR/debug
 CONFIG(release, debug|release) {
@@ -39,7 +36,7 @@
     QMAKE_POST_LINK += bash postScript_mac.sh "$$DESTDIR" "$$SANKORE_DIR/$$BUILD_DIR/product/Open-Sankore.app/Contents/MacOS"
 }
 
-LIBS     += "-L$$THIRD_PARTY_PATH/quazip/lib/$$SUB_DIR" "-lquazip"
+LIBS     += "-lquazip1-qt5"
 
 QT       += xml xmlpatterns core
 QT       += gui
@@ -51,9 +48,7 @@
 
 INCLUDEPATH += src
 
-DEPENDPATH  += $$THIRD_PARTY_PATH/quazip/
-INCLUDEPATH += $$THIRD_PARTY_PATH/quazip/
-include($$THIRD_PARTY_PATH/quazip/quazip.pri)
+INCLUDEPATH += /usr/include/QuaZip-Qt5-1.1/quazip
 
 
 SOURCES += \
--- OpenBoard-1.5.4.orig/libs.pri	2020-10-11 14:17:54.587596163 +0200
+++ OpenBoard-1.5.4/libs.pri	2020-10-11 14:13:49.516394503 +0200
@@ -23,7 +23,7 @@
 }
 
 QUAZIP_DIR   = "/usr"
-LIBS        += "-lquazip5"
+LIBS        += "-lquazip1-qt5"
 
 LIBS        += "-lpoppler"
 INCLUDEPATH += "/usr/include/poppler"
