--- db/nist/CMakeLists.txt
+++ db/nist/CMakeLists.txt
@@ -10,7 +10,7 @@ set(NIST_SAMPLE_MODELS
 )
 
 foreach(im ${NIST_SAMPLE_MODELS})
-  ADD_G_TARGET(${im} "BRL-CAD Geometry Models/Sample")
+#  ADD_G_TARGET(${im} "BRL-CAD Geometry Models/Sample")
 endforeach(im ${NIST_SAMPLE_MODELS})
 
 CMAKEFILES(
--- regress/nurbs/CMakeLists.txt
+++ regress/nurbs/CMakeLists.txt
@@ -35,11 +35,11 @@ add_custom_target(regress-nurbs)
 #set_target_properties(regress-nurbs PROPERTIES EXCLUDE_FROM_DEFAULT_BUILD 1)
 set_target_properties(regress-nurbs PROPERTIES FOLDER "BRL-CAD Regression Tests")
 
-ADD_NURBS_NIRT_Test(NIST03_MISS_01
-  "-M -b"
-  "${DATA_DIR}/db/nist/NIST_MBE_PMI_3.g"
-  "Document -114.61944548140064626 -127.49462006675899772 -134.83948901109894791 35 25"
-  "analyze_nhit;NIST_MBE_PMI_3.g")
+#ADD_NURBS_NIRT_Test(NIST03_MISS_01
+#  "-M -b"
+#  "${DATA_DIR}/db/nist/NIST_MBE_PMI_3.g"
+#  "Document -114.61944548140064626 -127.49462006675899772 -134.83948901109894791 35 25"
+#  "analyze_nhit;NIST_MBE_PMI_3.g")
 
 #ADD_NURBS_NIRT_Test(PLATE_MODE_CYL_HIT_01
 #  "-b"
