diff --git a/tests/test_license.py b/tests/test_license.py
index 093fa2c..340add7 100644
--- a/tests/test_license.py
+++ b/tests/test_license.py
@@ -8,4 +8,4 @@ def test_make_text():
     assert 'beginOptional' not in license.make_text()
     assert license.make_text().strip().startswith('MIT License')
 
-    assert 'Copyright (c) 2018 Gram' in license.make_text(copyright='2018 Gram')
+    assert 'MIT License 2018 Gram' in license.make_text(copyright='2018 Gram')
