diff --git a/PKGBUILD b/PKGBUILD
index 7ad438e..e191345 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -38,12 +38,14 @@ prepare() {
 }
 
 build() {
+  export GITHUB_PAT=dummy
+  mkdir -p ~/.R && echo "CXXFLAGS += -Wno-template-body" >> ~/.R/Makevars
   export CMAKE_C_COMPILER_LAUNCHER=ccache
   export CMAKE_CXX_COMPILER_LAUNCHER=ccache
   export CMAKE_Fortran_COMPILER_LAUNCHER=ccache
 
   export CFLAGS="${CFLAGS} -fPIC"
-  export CXXFLAGS="${CXXFLAGS} -fPIC"
+  export CXXFLAGS="${CXXFLAGS} -fPIC -Wno-template-body"
   export FCFLAGS="${FCFLAGS} -fPIC"
   export FFLAGS="${FFLAGS} -fPIC"
 
