diff -aur pythia8235-old/Makefile pythia8235-new/Makefile
--- pythia8235-old/Makefile	2018-11-28 00:54:49.657105842 +0000
+++ pythia8235-new/Makefile	2018-11-28 00:54:26.030804438 +0000
@@ -134,7 +134,7 @@
 $(LOCAL_LIB)/pythia8.py: $(LOCAL_INCLUDE)/Pythia8Plugins/PythonWrapper.h
 	SPLIT=`grep -n "PYTHON SOURCE" $< | cut -d : -f 1`;\
 	 SPLIT=$$[$$SPLIT+1]; tail -n +$$SPLIT $< | cut -d "/" -f 3- > $@
-	$(PYTHON_BIN)python -m compileall $(LOCAL_LIB)
+	$(PYTHON_BIN)python2 -m compileall $(LOCAL_LIB)
 $(LOCAL_LIB)/_pythia8.so: $(LOCAL_INCLUDE)/Pythia8Plugins/PythonWrapper.h\
 	$(LOCAL_LIB)/pythia8.py $(wildcard $(LOCAL_INCLUDE)/*/*.h) |\
 	$(LOCAL_LIB)/libpythia8$(LIB_SUFFIX)
