diff -aur pythia8235-old/Makefile pythia8235-new/Makefile
--- pythia8235-old/Makefile	2018-03-27 18:15:17.000000000 +0100
+++ pythia8235-new/Makefile	2018-11-27 23:42:13.813863630 +0000
@@ -102,7 +102,7 @@
 	$(CXX) $< -o $@ -c $(OBJ_COMMON)
 $(LOCAL_LIB)/libpythia8.a: $(OBJECTS)
 	rm -f $(LOCAL_LIB)/libpythia8$(LIB_SUFFIX)
-	ar cru $@ $^
+	ar rcs $@ $^
 $(LOCAL_LIB)/libpythia8$(LIB_SUFFIX): $(OBJECTS)
 	$(CXX) $^ -o $@ $(CXX_COMMON) $(CXX_SHARED) $(CXX_SONAME)$(notdir $@)\
 	  $(LIB_COMMON)
