diff -u org/setup.py new/setup.py
--- org/setup.py	2022-06-17 17:48:47.000000000 +0200
+++ new/setup.py	2022-07-06 22:22:36.633826832 +0200
@@ -314,7 +314,7 @@
         and os.path.exists(os.path.join(CUDA_HOME, "include", "nvjpeg.h"))
     )
 
-    print(f"NVJPEG found: {nvjpeg_found}")
+    nvjpeg_found = False; print(f"NVJPEG found: {nvjpeg_found}")
     image_macros += [("NVJPEG_FOUND", str(int(nvjpeg_found)))]
     if nvjpeg_found:
         print("Building torchvision with NVJPEG image support")
