diff -aur xrootd-4.8.3-old/src/XrdApps/XrdCpFile.cc xrootd-4.8.3-new/src/XrdApps/XrdCpFile.cc
--- xrootd-4.8.3-old/src/XrdApps/XrdCpFile.cc	2018-05-13 19:01:03.602433537 +0200
+++ xrootd-4.8.3-new/src/XrdApps/XrdCpFile.cc	2018-05-13 19:01:48.269454421 +0200
@@ -88,7 +88,7 @@
    for (i = 0; i < pTnum; i++)
        {if (!strncmp(FSpec, pTab[i].pHdr, pTab[i].pHsz))
            {Protocol = pTab[i].pVal;
-            strncpy(ProtName, pTab[i].pHdr, pTab[i].pHsz-3);
+            memcpy(ProtName, pTab[i].pHdr, pTab[i].pHsz-3);
             return;
            }
        }
