diff -aur xrootd-4.6.0-old/src/XrdCns/XrdCnsSsi.cc xrootd-4.6.0-new/src/XrdCns/XrdCnsSsi.cc
--- xrootd-4.6.0-old/src/XrdCns/XrdCnsSsi.cc	2017-02-06 17:12:16.000000000 +0100
+++ xrootd-4.6.0-new/src/XrdCns/XrdCnsSsi.cc	2017-05-30 14:54:32.220679575 +0200
@@ -60,7 +60,9 @@
 {
 char Info[XrdCnsLogRec::FixDLen];
 
-void Updt(const char *nInfo) {strncpy(Info, nInfo, sizeof(Info));}
+void Updt(const char *nInfo) {
+     if(nInfo == NULL) nInfo = "";
+     strncpy(Info, nInfo, sizeof(Info));}
 
      XrdCnsSsiFRec(const char *Data) {if (!Data) Data = XrdCnsLogRec::iArg;
                                       strncpy(Info, Data, sizeof(Info));
diff -aur xrootd-4.6.0-old/src/XrdXrootd/XrdXrootdJob.cc xrootd-4.6.0-new/src/XrdXrootd/XrdXrootdJob.cc
--- xrootd-4.6.0-old/src/XrdXrootd/XrdXrootdJob.cc	2017-02-06 17:12:16.000000000 +0100
+++ xrootd-4.6.0-new/src/XrdXrootd/XrdXrootdJob.cc	2017-05-30 14:54:32.224013013 +0200
@@ -324,6 +324,7 @@
    for (i = 0; i < numClients; i++)
        if (!Client[i].Link->isInstance(Client[i].Inst))
           {k = i;
+           if(i == numClients - 1) continue;
            for (j = i+1; j < numClients; j++,k++) Client[k] = Client[j];
            numClients--; i--;
           }
