--- src/RapidOCR-3.8.4/python/pyproject.toml	2026-06-18 16:02:34.692772070 -0400
+++ src/RapidOCR-3.8.4/python/pyprojectd.toml	2026-06-18 16:03:23.963705999 -0400
@@ -2,13 +2,13 @@
 requires = [
     "setuptools>=77",
     "wheel",
-    "setuptools-scm>=8",
 ]
 build-backend = "setuptools.build_meta"
 
 [project]
 name = "rapidocr"
-dynamic = ["version", "dependencies"]
+dynamic = ["dependencies"]
+version = "3.8.4"
 description = "Awesome OCR Library"
 readme = { file = "README.md", content-type = "text/markdown" }
 requires-python = ">=3.8,<4"
