From: Sophie Brun <sophie@freexian.com>
Date: Tue, 5 Nov 2019 10:39:14 +0100
Subject: Fix spelling errors

Last-Update: 2018-07-12
---
 sources/pyside2/libpyside/pysideproperty.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sources/pyside2/libpyside/pysideproperty.cpp b/sources/pyside2/libpyside/pysideproperty.cpp
index 0d0957f..86909d3 100644
--- a/sources/pyside2/libpyside/pysideproperty.cpp
+++ b/sources/pyside2/libpyside/pysideproperty.cpp
@@ -519,7 +519,7 @@ int setValue(PySideProperty *self, PyObject *source, PyObject *value)
         Shiboken::AutoDecRef result(PyObject_CallObject(fdel, args));
         return (result.isNull() ? -1 : 0);
     }
-    PyErr_SetString(PyExc_AttributeError, "Attibute read only");
+    PyErr_SetString(PyExc_AttributeError, "Attribute read only");
     return -1;
 }
 
