--- a/bin/OpenGrok	2016-11-06 16:47:25.000000000 +0000
+++ b/bin/OpenGrok	2016-11-06 16:35:46.335876574 +0000
@@ -122,9 +122,7 @@
 	exit 1
 fi
 
-# TODO: Handle symlinks correctly (especially in ${0})
-SCRIPT_DIRECTORY="`dirname ${0}`"
-SCRIPT_DIRECTORY="`cd ${SCRIPT_DIRECTORY}; pwd`"
+SCRIPT_DIRECTORY="/usr/share/java/opengrok"
 
 #
 # Default Instance Configuration
@@ -164,8 +162,8 @@
             LOGGER_CONF_SOURCE="${SCRIPT_DIRECTORY}/${LOGGER_CONFIG_FILE}"
         else
             # Handle Binary Distributions
-            OPENGROK_DISTRIBUTION_BASE="${SCRIPT_DIRECTORY}/../lib"
-            LOGGER_CONF_SOURCE="${OPENGROK_DISTRIBUTION_BASE}/../doc/"
+            OPENGROK_DISTRIBUTION_BASE="${SCRIPT_DIRECTORY}"
+            LOGGER_CONF_SOURCE="${OPENGROK_DISTRIBUTION_BASE}/doc/"
             LOGGER_CONF_SOURCE="${LOGGER_CONF_SOURCE}${LOGGER_CONFIG_FILE}"
         fi
     fi
@@ -225,12 +223,7 @@
         CTAGS_OPTIONS_FILE="${OPENGROK_INSTANCE_BASE}/etc/ctags.config"
     fi
 
-    # REQUIRED: Java Home
-    JAVA_HOME="${JAVA_HOME:-`FindJavaHome`}"
-    export JAVA_HOME
-
-    # REQUIRED: Java Virtual Machine
-    JAVA="${JAVA:-$JAVA_HOME/bin/java}"
+    JAVA="/usr/bin/java"
 
     # DEVELOPMENT: Debug option, if enabled current indexer will listen on the
     # port 8010 until a debugger connects
@@ -544,6 +537,7 @@
     if [    -d "/var/tomcat6/webapps"        \
          -o -d "/var/lib/tomcat6/webapps"    \
          -o -d "/var/lib/tomcat7/webapps"    \
+         -o -d "/var/lib/tomcat8/webapps"    \
        ]
     then
         echo "Tomcat"
@@ -576,7 +570,8 @@
         ${OPENGROK_TOMCAT_BASE} \
         /var/tomcat6            \
         /var/lib/tomcat6        \
-        /var/lib/tomcat7
+        /var/lib/tomcat7        \
+        /var/lib/tomcat8
     do
         if [ -d "${prefix}/webapps" ]
         then
