diff --git a/CMakeLists.txt b/CMakeLists.txt
index e1f7f0938..267995965 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.13)
 
 if (POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
diff --git a/cmake/SymEngineConfig.cmake.in b/cmake/SymEngineConfig.cmake.in
index 5b5533ab8..1e4f79552 100644
--- a/cmake/SymEngineConfig.cmake.in
+++ b/cmake/SymEngineConfig.cmake.in
@@ -31,7 +31,7 @@
 # target_link_libraries(example ${SYMENGINE_LIBRARIES})
 #
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.13)
 
 if (POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
