diff --git a/cmake/FindARB.cmake b/cmake/FindARB.cmake
index 281c5bea..f0428e9c 100644
--- a/cmake/FindARB.cmake
+++ b/cmake/FindARB.cmake
@@ -1,7 +1,7 @@
 include(LibFindMacros)
 
-libfind_include(arb.h arb)
-libfind_library(arb arb flint-arb)
+libfind_include(flint/arb.h arb)
+libfind_library(arb arb flint-arb flint)
 
 set(ARB_LIBRARIES ${ARB_LIBRARY})
 set(ARB_INCLUDE_DIRS ${ARB_INCLUDE_DIR})
diff --git a/symengine/eval_arb.h b/symengine/eval_arb.h
index 8d6f00c1..e162a9ff 100644
--- a/symengine/eval_arb.h
+++ b/symengine/eval_arb.h
@@ -11,7 +11,8 @@
 
 #ifdef HAVE_SYMENGINE_ARB
 #include <symengine/basic.h>
-#include <arb.h>
+#include <mpfr.h>
+#include <flint/arb.h>
 
 namespace SymEngine
 {
diff --git a/symengine/flint_wrapper.h b/symengine/flint_wrapper.h
index 21cfcef0..edf07e22 100644
--- a/symengine/flint_wrapper.h
+++ b/symengine/flint_wrapper.h
@@ -5,6 +5,8 @@
 #include <gmp.h>
 
 #include <flint/fmpz.h>
+#include <flint/fmpz_poly.h>
+#include <flint/fmpz_poly_factor.h>
 #include <flint/fmpq.h>
 #include <flint/fmpq_poly.h>
 
@@ -708,15 +710,15 @@ public:
         fmpq_poly_init(poly);
         fmpq_poly_set_str(poly, cp);
     }
-    fmpq_poly_wrapper(const mpz_t z)
+    fmpq_poly_wrapper(const fmpz_t z)
     {
         fmpq_poly_init(poly);
-        fmpq_poly_set_mpz(poly, z);
+        fmpq_poly_set_fmpz(poly, z);
     }
-    fmpq_poly_wrapper(const mpq_t q)
+    fmpq_poly_wrapper(const fmpq_t q)
     {
         fmpq_poly_init(poly);
-        fmpq_poly_set_mpq(poly, q);
+        fmpq_poly_set_fmpq(poly, q);
     }
     fmpq_poly_wrapper(const fmpq_wrapper &q)
     {
diff --git a/symengine/ntheory.cpp b/symengine/ntheory.cpp
index 6108bf2c..7c6d08a1 100644
--- a/symengine/ntheory.cpp
+++ b/symengine/ntheory.cpp
@@ -14,9 +14,10 @@
 #include <primesieve.hpp>
 #endif // HAVE_SYMENGINE_PRIMESIEVE
 #ifdef HAVE_SYMENGINE_ARB
-#include "arb.h"
-#include "bernoulli.h"
+#include <flint/arb.h>
+#include <flint/bernoulli.h>
 #include "rational.h"
+#include <flint/fmpq.h>
 #endif // HAVE_SYMENGINE_ARB
 #ifndef HAVE_SYMENGINE_GMP
 #include <boost/random/uniform_int.hpp>
diff --git a/symengine/tests/eval/test_eval_arb.cpp b/symengine/tests/eval/test_eval_arb.cpp
index df9bb61f..0eecf647 100644
--- a/symengine/tests/eval/test_eval_arb.cpp
+++ b/symengine/tests/eval/test_eval_arb.cpp
@@ -947,4 +947,4 @@ TEST_CASE("Zeta: eval_arb", "[eval_arb]")
     REQUIRE(arb_contains_mpfr(a, f));
     mpfr_clear(f);
     arb_clear(a);
-}
\ No newline at end of file
+}
diff --git a/symengine/utilities/catch/catch.cpp b/symengine/utilities/catch/catch.cpp
index 7291ac5..d0db607 100644
--- a/symengine/utilities/catch/catch.cpp
+++ b/symengine/utilities/catch/catch.cpp
@@ -13,7 +13,7 @@
 #endif // HAVE_SYMENGINE_MPFR
 
 #if defined(HAVE_SYMENGINE_ARB)
-#include <arb.h>
+#include <flint/arb.h>
 #endif // HAVE_SYMENGINE_ARB
 
 using SymEngine::print_stack_on_segfault;
