From f758974a6f6104020e1f34cfe678a8fd99385c12 Mon Sep 17 00:00:00 2001
From: fdhaussy <florian.dhaussy@protonmail.com>
Date: Fri, 7 Oct 2022 16:02:26 +0200
Subject: [PATCH] fix: expired url in test file

---
 tests/test_finder.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/tests/test_finder.py b/tests/test_finder.py
index aa52235..77c7846 100644
--- a/tests/test_finder.py
+++ b/tests/test_finder.py
@@ -29,10 +29,6 @@ from dephell_changelogs._finder import get_changelog_url
         'selenium',
         'https://raw.githubusercontent.com/SeleniumHQ/selenium/master/py/CHANGES',
     ),
-    (
-        'sqlalchemy',
-        'https://raw.githubusercontent.com/sqlalchemy/sqlalchemy/master/CHANGES',
-    ),
     (
         'django-hijack',
         'https://raw.githubusercontent.com/arteria/django-hijack/master/CHANGELOG.txt',
-- 
2.37.3

