--- ParaView-v5.7.0/VTK/ThirdParty/pugixml/CMakeLists.txt.orig	2019-10-18 16:22:47.612817047 +0800
+++ ParaView-v5.7.0/VTK/ThirdParty/pugixml/CMakeLists.txt	2019-10-18 16:29:12.692090865 +0800
@@ -1,7 +1,12 @@
-vtk_module_third_party_internal(
-  LICENSE_FILES "vtkpugixml/LICENSE"
-  VERSION       "1.9"
-  STANDARD_INCLUDE_DIRS)
+vtk_module_third_party(
+  INTERNAL
+    LICENSE_FILES "vtkpugixml/LICENSE"
+    VERSION       "1.9"
+    STANDARD_INCLUDE_DIRS
+  EXTERNAL
+    PACKAGE pugixml
+    TARGETS pugixml
+    STANDARD_INCLUDE_DIRS)
 
 configure_file(
   "${CMAKE_CURRENT_SOURCE_DIR}/vtk_pugixml.h.in"
