--- a/bazel/ray_deps_setup.bzl
+++ b/bazel/ray_deps_setup.bzl
@@ -453,6 +453,7 @@
         build_file = "@io_ray//bazel:jemalloc.BUILD",
         sha256 = "2db82d1e7119df3e71b7640219b6dfe84789bc0537983c3b7ac4f7189aecfeaa",
         strip_prefix = "jemalloc-5.3.0",
+        patch_cmds = ["sed -i 's/std::__throw_bad_alloc()/throw std::bad_alloc()/' src/jemalloc_cpp.cpp"],
     )
 
 
--- a/src/ray/observability/open_telemetry_metric_recorder.h
+++ b/src/ray/observability/open_telemetry_metric_recorder.h
@@ -14,6 +14,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <opentelemetry/exporters/otlp/otlp_grpc_metric_exporter.h>
 #include <opentelemetry/metrics/meter.h>
 #include <opentelemetry/metrics/observer_result.h>
--- a/src/ray/thirdparty/dlmalloc.c
+++ b/src/ray/thirdparty/dlmalloc.c
@@ -1,3 +1,4 @@
+#pragma GCC diagnostic ignored "-Wunused-but-set-variable"
 /*
   This is a version (aka dlmalloc) of malloc/free/realloc written by
   Doug Lea and released to the public domain, as explained at
