--- a/cmake/gazebo-config.cmake.in
+++ b/cmake/gazebo-config.cmake.in
@@ -160,7 +160,7 @@
   "${OGRE_INSTALL_PREFIX}/share/OGRE/cmake/modules;${OGRE_INSTALL_PREFIX}/lib/OGRE/cmake;${OGRE_INSTALL_PREFIX}/CMake;${CMAKE_MODULE_PATH}")
 
 # Find boost
-find_package(Boost @MIN_BOOST_VERSION@ REQUIRED thread system
+find_package(Boost @MIN_BOOST_VERSION@ REQUIRED thread
   filesystem program_options regex iostreams date_time)
 list(APPEND @PKG_NAME@_INCLUDE_DIRS ${Boost_INCLUDE_DIRS})
 list(APPEND @PKG_NAME@_LIBRARIES ${Boost_LIBRARIES})
