diff --git a/CMakeLists.txt b/CMakeLists.txt
index 57af9375..f41a8367 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -370,7 +370,7 @@ else()
 	# https://youtu.be/bsXLMQ6WgIk).
 	# By using CMAKE_CXX_STANDARD we also avoid having to implement
 	# compiler specific command line arguments.
-	set(CMAKE_CXX_STANDARD 11)
+	set(CMAKE_CXX_STANDARD 14)
 	# do not fall back to a previous standard
 	set(CMAKE_CXX_STANDARD_REQUIRED ON)
 	# enable compiler extensions like -std=gnu++11 on gcc
diff --git a/src/spherical_quadtree/spherical_quadtree.cc b/src/spherical_quadtree/spherical_quadtree.cc
index 6b7c08bd..c7e3f38d 100644
--- a/src/spherical_quadtree/spherical_quadtree.cc
+++ b/src/spherical_quadtree/spherical_quadtree.cc
@@ -6,21 +6,6 @@
 
 std::mt19937 gen(0);
 
-#ifndef _MSC_VER
-// needed until we compile with C++14
-#if __cplusplus >= 201402L
-#error remove definition of make_unique
-#endif
-namespace std
-{
-template <typename T, typename... Args>
-std::unique_ptr<T> make_unique(Args &&... args)
-{
-	return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
-}
-} // namespace std
-#endif
-
 static void circumcircle(const double *v1, const double *v2, const double *v3,
 			 double *p, double *theta)
 {
