--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -103,7 +103,11 @@
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-sign-compare -O3 ${ENV_ARCHFLAGS}")
     endif()
 else()
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-sign-compare -O3 -DROOT_PATH='\"${TOPLEVEL_PREFIX_PATH}\"' ${ENV_ARCHFLAGS}")
+  if(NOT DEFINED DOCLING_PARSE_ROOT_PATH)
+    set(DOCLING_PARSE_ROOT_PATH "${TOPLEVEL_PREFIX_PATH}")
+  endif()
+
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-sign-compare -O3 -DROOT_PATH='\"${DOCLING_PARSE_ROOT_PATH}\"' ${ENV_ARCHFLAGS}")
   # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-sign-compare -g3 -O0 -fno-omit-frame-pointer -DROOT_PATH='\"${TOPLEVEL_PREFIX_PATH}\"' ${ENV_ARCHFLAGS}")
 endif()
