Index: VTK-7.0.0/Examples/GUI/Qt/GraphicsView/CMakeLists.txt
===================================================================
--- VTK-7.0.0.orig/Examples/GUI/Qt/GraphicsView/CMakeLists.txt
+++ VTK-7.0.0/Examples/GUI/Qt/GraphicsView/CMakeLists.txt
@@ -80,6 +80,6 @@ else()
     qt4_wrap_cpp(MOC_Srcs ${MOC_Hdrs})
 
     add_executable(qtgraphicsview ${Srcs} ${Hdrs} ${MOC_Hdrs} ${QRC_Srcs})
-    target_link_libraries(qtgraphicsview ${QT_LIBRARIES} ${VTK_LIBRARIES})
+    target_link_libraries(qtgraphicsview Qt4::QtCore Qt4::QtOpenGL Qt4::QtWebKit Qt4::QtGui ${VTK_LIBRARIES})
   endif()
 endif()
