diff --git a/PKGBUILD b/PKGBUILD
index 4556840..1b1311b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,12 +20,13 @@ _dir="gz-common-ignition-common3_${pkgver}"
 
 build() {
   cd "$srcdir/$_dir"
+  sed -i '1i #include <cstdint>' profiler/include/gz/common/Profiler.hh
 
   mkdir -p build
   cd build
 
   export PKG_CONFIG_PATH=/usr/lib/ffmpeg4.4/pkgconfig
-  cmake .. -DCMAKE_BUILD_TYPE="Release" \
+  cmake .. -DCMAKE_BUILD_TYPE="Release" -DCMAKE_CXX_FLAGS="-fpermissive" \
            -DCMAKE_INSTALL_PREFIX="/usr" \
            -DCMAKE_INSTALL_LIBDIR="lib" \
            -DBUILD_TESTING:BOOL=False
