--- libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h    2020-02-28 11:49:29.763277856 +0000
+++ libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h    2020-02-28 11:49:45.092909823 +0000
@@ -198,8 +198,7 @@
     unsigned long long __unused1;
     unsigned long long __unused2;
 #else
-    unsigned short mode;
-    unsigned short __pad1;
+    unsigned int mode;
     unsigned short __seq;
     unsigned short __pad2;
 #if defined(__x86_64__) && !defined(_LP64)