diff -Naurp a/svm-toy/qt/Makefile b/svm-toy/qt/Makefile
--- a/svm-toy/qt/Makefile	2018-12-04 18:34:40.715220500 +0000
+++ b/svm-toy/qt/Makefile	2018-12-04 18:45:00.689709128 +0000
@@ -1,7 +1,7 @@
 # use  ``export QT_SELECT=qt5'' in a command window for using qt5
 # may need to adjust the path of header files
 CXX? = g++
-INCLUDE = /usr/include/x86_64-linux-gnu/qt5
+INCLUDE = /usr/include/qt
 CFLAGS = -Wall -O3 -I$(INCLUDE) -I$(INCLUDE)/QtWidgets -I$(INCLUDE)/QtGui -I$(INCLUDE)/QtCore -fPIC -std=c++11
 LIB = -lQt5Widgets -lQt5Gui -lQt5Core
 MOC = /usr/bin/moc
