--- src/OpenFOAM-3.0.x/bin/paraFoam
+++ src/OpenFOAM-3.0.x/bin/paraFoam
@@ -124,21 +124,22 @@ done
 # Get the ParaView major version to select the appropriate readers
 version=`echo $ParaView_VERSION | sed -e 's/^\([0-9][0-9]*\).*$/\1/'`

-# Check that reader module has been built
-if [ $requirePV -eq 1 -a ! -f $PV_PLUGIN_PATH/libPV${version}FoamReader_SM.so ]
-then
-    cat<< BUILDREADER
+# NOTE: For OpenFOAM built from AUR, paraview will already be installed
+# with the required readers. No need to perform this check.
+#if [ $requirePV -eq 1 -a ! -f $PV_PLUGIN_PATH/libPV${version}FoamReader_SM.so ]
+#then
+#    cat<< BUILDREADER

-FATAL ERROR: ParaView reader module libraries do not exist
+#FATAL ERROR: ParaView reader module libraries do not exist

-Please build the reader module before continuing:
-cd \$FOAM_UTILITIES/postProcessing/graphics/PV${version}Readers
-./Allwclean
-./Allwmake
+#Please build the reader module before continuing:
+#cd \$FOAM_UTILITIES/postProcessing/graphics/PV${version}Readers
+#./Allwclean
+#./Allwmake

-BUILDREADER
-    exit 1
-fi
+#BUILDREADER
+#    exit 1
+#fi

 # Check for --data=... argument
 hasDataArg()
