From 60e8045fc9d3dabf6a941800ee70e06ccc19a2dc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20La=C3=9F?= <bevan@bi-co.net>
Date: Thu, 9 Oct 2025 20:08:20 +0200
Subject: [PATCH 4/5] Update easybind to avoid using unavailable snapshot

---
 build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index 70d35b7c64..c00ad16ed6 100644
--- a/build.gradle
+++ b/build.gradle
@@ -221,7 +221,7 @@ dependencies {
     implementation 'org.kordamp.ikonli:ikonli-materialdesign2-pack:12.3.1'
     implementation 'com.github.sialcasa.mvvmFX:mvvmfx-validation:f195849ca9' //jitpack
     implementation 'de.saxsys:mvvmfx:1.8.0'
-    implementation('com.tobiasdiez:easybind:2.2.1-SNAPSHOT') {
+    implementation('org.jabref:easybind:2.3.0') {
         exclude group: 'org.openjfx'
     }
     implementation 'org.fxmisc.flowless:flowless:0.7.3'
-- 
2.53.0

