--- a/libcelt/celt.c
+++ b/libcelt/celt.c
@@ -2692,7 +2692,7 @@ const char *celt_strerror(int error)
    static const char *error_strings[8] = {
       "success",
       "invalid argument",
-      "invalid mode",
+      "buffer too small",
       "internal error",
       "corrupted stream",
       "request not implemented",
--- a/tests/tandem-test.c
+++ b/tests/tandem-test.c
@@ -112,7 +112,7 @@ int async_tandem(int rate, int frame_size, int channels, int bitrate_min,
             }
 
             ret = celt_decode(dec, data, ret, pcm, frame_size);
-            if (ret != CELT_OK) {
+            if (ret < 0) {
                 fprintf(stderr, "Error: celt_decode returned %s\n", celt_strerror(ret));
             }
         }
@@ -137,7 +137,7 @@ int async_tandem(int rate, int frame_size, int channels, int bitrate_min,
             }
 
             ret = celt_decode(dec, data, ret, pcm, frame_size);
-            if (ret != CELT_OK) {
+            if (ret < 0) {
                 fprintf(stderr, "Error: at %d bytes_per_frame celt_decode returned %s\n",
                         bytes_per_frame, celt_strerror(ret));
                 exit(1);
