diff --unified --recursive --text src.orig/saga-8.5.0/saga-gis/CMakeLists.txt src.new/saga-8.5.0/saga-gis/CMakeLists.txt
--- src.orig/saga-8.5.0/saga-gis/CMakeLists.txt	2022-12-20 16:41:11.000000000 +0100
+++ src.new/saga-8.5.0/saga-gis/CMakeLists.txt	2023-01-25 17:33:10.601703818 +0100
@@ -53,7 +53,7 @@
 include(version.cmake)
 
 # include shared definitions and scripts
-include(cmake/shared/require-cpp11.cmake)
+# include(cmake/shared/require-cpp11.cmake)
 include(cmake/shared/enable_warnings.cmake)
 include(cmake/shared/windows_version_info.cmake)
 
