From 7c6726d975d54185d1318a112059f3e9ca5edf36 Mon Sep 17 00:00:00 2001
From: Ranieri Althoff <1993083+ranisalt@users.noreply.github.com>
Date: Sat, 18 Apr 2020 18:02:58 +0000
Subject: [PATCH] Install amdocl64.icd where ICD loader expects it

---
 api/opencl/amdocl/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/api/opencl/amdocl/CMakeLists.txt b/api/opencl/amdocl/CMakeLists.txt
index e217110..2731a6a 100644
--- rocm-opencl-runtime/api/opencl/amdocl/CMakeLists.txt
+++ rocm-opencl-runtime/api/opencl/amdocl/CMakeLists.txt
@@ -83,3 +83,6 @@ if(${USE_COMGR_LIBRARY} MATCHES "yes")
 else()
   target_link_libraries(amdocl64 opencl_driver oclelf pthread dl ${ROCT_LIBRARIES} ${ROCR_LIBRARIES})
 endif()
+
+file(GENERATE OUTPUT "${CMAKE_BINARY_DIR}/amdocl64.icd" CONTENT "$<TARGET_FILE_NAME:amdocl64>")
+install(FILES "${CMAKE_BINARY_DIR}/amdocl64.icd" DESTINATION "${CMAKE_INSTALL_SYSCONFDIR}/OpenCL/vendors/")
