--- a/cmake/DARTFindEigen3.cmake	2024-11-15 08:57:40.000000000 -0800
+++ b/cmake/DARTFindEigen3.cmake	2026-05-03 14:05:53.641391646 -0700
@@ -6,4 +6,4 @@
 #
 # This file is provided under the "BSD-style" License
 
-find_package(Eigen3 3.4.0 REQUIRED CONFIG)
+find_package(Eigen3 REQUIRED CONFIG)
--- a/dart/common/detail/Logging-impl.hpp	2024-11-15 08:57:40.000000000 -0800
+++ b/dart/common/detail/Logging-impl.hpp	2026-05-03 14:35:32.553631258 -0700
@@ -80,7 +80,7 @@
 void trace(const S& format_str, [[maybe_unused]] Args&&... args)
 {
 #if DART_HAVE_spdlog
-  spdlog::trace(format_str, std::forward<Args>(args)...);
+  spdlog::trace(fmt::runtime(format_str), std::forward<Args>(args)...);
 #else
   detail::print(
       std::cout, "[trace]", format_str, 38, std::forward<Args>(args)...);
@@ -92,7 +92,7 @@
 void debug(const S& format_str, [[maybe_unused]] Args&&... args)
 {
 #if DART_HAVE_spdlog
-  spdlog::debug(format_str, std::forward<Args>(args)...);
+  spdlog::debug(fmt::runtime(format_str), std::forward<Args>(args)...);
 #else
   detail::print(
       std::cout, "[debug]", format_str, 36, std::forward<Args>(args)...);
@@ -104,7 +104,7 @@
 void info(const S& format_str, [[maybe_unused]] Args&&... args)
 {
 #if DART_HAVE_spdlog
-  spdlog::info(format_str, std::forward<Args>(args)...);
+  spdlog::info(fmt::runtime(format_str), std::forward<Args>(args)...);
 #else
   detail::print(
       std::cout, "[info]", format_str, 32, std::forward<Args>(args)...);
@@ -116,7 +116,7 @@
 void warn(const S& format_str, [[maybe_unused]] Args&&... args)
 {
 #if DART_HAVE_spdlog
-  spdlog::warn(format_str, std::forward<Args>(args)...);
+  spdlog::warn(fmt::runtime(format_str), std::forward<Args>(args)...);
 #else
   detail::print(
       std::cerr, "[warn]", format_str, 33, std::forward<Args>(args)...);
@@ -128,7 +128,7 @@
 void error(const S& format_str, [[maybe_unused]] Args&&... args)
 {
 #if DART_HAVE_spdlog
-  spdlog::error(format_str, std::forward<Args>(args)...);
+  spdlog::error(fmt::runtime(format_str), std::forward<Args>(args)...);
 #else
   detail::print(
       std::cerr, "[error]", format_str, 31, std::forward<Args>(args)...);
@@ -140,7 +140,7 @@
 void fatal(const S& format_str, [[maybe_unused]] Args&&... args)
 {
 #if DART_HAVE_spdlog
-  spdlog::critical(format_str, std::forward<Args>(args)...);
+  spdlog::critical(fmt::runtime(format_str), std::forward<Args>(args)...);
 #else
   detail::print(
       std::cerr, "[fatal]", format_str, 35, std::forward<Args>(args)...);
