http://bugzilla.scilab.org/show_bug.cgi?id=13809
Index: scilab-5.5.2/configure
===================================================================
--- scilab-5.5.2.orig/configure
+++ scilab-5.5.2/configure
@@ -14264,7 +14264,7 @@ $as_echo_n "checking jogl2... " >&6; }
 // #line 14264 "configure"
 import java.util.regex.Pattern;
 
-import javax.media.opengl.glu.GLUnurbs;
+import com.jogamp.opengl.glu.GLUnurbs;
 
 public class conftest {
     public static void main(String[] argv) {
@@ -14350,10 +14350,10 @@ $as_echo "$jar_resolved" >&6; }
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
       if test "" = "yes"; then
-         { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Could not find or use the Java package/jar jogl2 used by Scilab 3D rendering - Version 2.0 (looking for package javax.media.opengl.glu.GLUnurbs)" >&5
-$as_echo "$as_me: WARNING: Could not find or use the Java package/jar jogl2 used by Scilab 3D rendering - Version 2.0 (looking for package javax.media.opengl.glu.GLUnurbs)" >&2;}
+         { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Could not find or use the Java package/jar jogl2 used by Scilab 3D rendering - Version 2.0 (looking for package com.jogamp.opengl.glu.GLUnurbs)" >&5
+$as_echo "$as_me: WARNING: Could not find or use the Java package/jar jogl2 used by Scilab 3D rendering - Version 2.0 (looking for package com.jogamp.opengl.glu.GLUnurbs)" >&2;}
       else
-          as_fn_error $? "Could not find or use the Java package/jar jogl2 used by Scilab 3D rendering - Version 2.0 (looking for package javax.media.opengl.glu.GLUnurbs)" "$LINENO" 5
+          as_fn_error $? "Could not find or use the Java package/jar jogl2 used by Scilab 3D rendering - Version 2.0 (looking for package com.jogamp.opengl.glu.GLUnurbs)" "$LINENO" 5
       fi
     fi
 
Index: scilab-5.5.2/configure.ac
===================================================================
--- scilab-5.5.2.orig/configure.ac
+++ scilab-5.5.2/configure.ac
@@ -912,7 +912,7 @@ if test "$with_javasci" != no -o "$with_
                 AC_SUBST(SKINLF)
 
                 # JOGL 2
-                AC_JAVA_CHECK_PACKAGE([jogl2],[javax.media.opengl.glu.GLUnurbs],[Scilab 3D rendering - Version 2.0])
+                AC_JAVA_CHECK_PACKAGE([jogl2],[com.jogamp.opengl.glu.GLUnurbs],[Scilab 3D rendering - Version 2.0])
                 JOGL2=$PACKAGE_JAR_FILE
                 AC_SUBST(JOGL2)
 
Index: scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/SwingView.java
===================================================================
--- scilab-5.5.2.orig/modules/gui/src/java/org/scilab/modules/gui/SwingView.java
+++ scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/SwingView.java
@@ -84,10 +84,10 @@ import java.util.Map;
 import java.util.Set;
 import java.util.UUID;
 
-import javax.media.opengl.GL;
-import javax.media.opengl.GLCapabilities;
-import javax.media.opengl.GLProfile;
-import javax.media.opengl.awt.GLCanvas;
+import com.jogamp.opengl.GL;
+import com.jogamp.opengl.GLCapabilities;
+import com.jogamp.opengl.GLProfile;
+import com.jogamp.opengl.awt.GLCanvas;
 import javax.swing.JComponent;
 import javax.swing.JSeparator;
 import javax.swing.SwingUtilities;
Index: scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/bridge/ScilabBridge.java
===================================================================
--- scilab-5.5.2.orig/modules/gui/src/java/org/scilab/modules/gui/bridge/ScilabBridge.java
+++ scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/bridge/ScilabBridge.java
@@ -20,7 +20,7 @@ import java.awt.Cursor;
 import java.awt.Font;
 import java.awt.image.BufferedImage;
 
-import javax.media.opengl.GL;
+import com.jogamp.opengl.GL;
 
 import org.scilab.modules.gui.canvas.Canvas;
 import org.scilab.modules.gui.canvas.ScilabCanvas;
Index: scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/bridge/canvas/SwingScilabCanvas.java
===================================================================
--- scilab-5.5.2.orig/modules/gui/src/java/org/scilab/modules/gui/bridge/canvas/SwingScilabCanvas.java
+++ scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/bridge/canvas/SwingScilabCanvas.java
@@ -32,8 +32,8 @@ import java.awt.event.MouseListener;
 import java.awt.event.MouseMotionListener;
 import java.awt.image.BufferedImage;
 
-import javax.media.opengl.GL;
-import javax.media.opengl.GLAutoDrawable;
+import com.jogamp.opengl.GL;
+import com.jogamp.opengl.GLAutoDrawable;
 import javax.swing.JPanel;
 
 import org.scilab.forge.scirenderer.Canvas;
Index: scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/bridge/canvas/SwingScilabCanvasImpl.java
===================================================================
--- scilab-5.5.2.orig/modules/gui/src/java/org/scilab/modules/gui/bridge/canvas/SwingScilabCanvasImpl.java
+++ scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/bridge/canvas/SwingScilabCanvasImpl.java
@@ -19,12 +19,12 @@ import java.awt.HeadlessException;
 import java.util.Calendar;
 import java.util.StringTokenizer;
 
-import javax.media.opengl.GL;
-import javax.media.opengl.GLCapabilities;
-import javax.media.opengl.GLException;
-import javax.media.opengl.GLProfile;
-import javax.media.opengl.awt.GLCanvas;
-import javax.media.opengl.awt.GLJPanel;
+import com.jogamp.opengl.GL;
+import com.jogamp.opengl.GLCapabilities;
+import com.jogamp.opengl.GLException;
+import com.jogamp.opengl.GLProfile;
+import com.jogamp.opengl.awt.GLCanvas;
+import com.jogamp.opengl.awt.GLJPanel;
 
 import org.scilab.modules.action_binding.InterpreterManagement;
 import org.scilab.modules.commons.OS;
Index: scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/canvas/Canvas.java
===================================================================
--- scilab-5.5.2.orig/modules/gui/src/java/org/scilab/modules/gui/canvas/Canvas.java
+++ scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/canvas/Canvas.java
@@ -14,7 +14,7 @@ package org.scilab.modules.gui.canvas;
 
 import java.awt.image.BufferedImage;
 
-import javax.media.opengl.GL;
+import com.jogamp.opengl.GL;
 
 import org.scilab.modules.gui.dockable.Dockable;
 
Index: scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/canvas/ScilabCanvas.java
===================================================================
--- scilab-5.5.2.orig/modules/gui/src/java/org/scilab/modules/gui/canvas/ScilabCanvas.java
+++ scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/canvas/ScilabCanvas.java
@@ -16,7 +16,7 @@ package org.scilab.modules.gui.canvas;
 
 import java.awt.image.BufferedImage;
 
-import javax.media.opengl.GL;
+import com.jogamp.opengl.GL;
 
 import org.scilab.modules.gui.bridge.ScilabBridge;
 import org.scilab.modules.gui.dockable.ScilabDockable;
Index: scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/canvas/ScilabCanvasBridge.java
===================================================================
--- scilab-5.5.2.orig/modules/gui/src/java/org/scilab/modules/gui/canvas/ScilabCanvasBridge.java
+++ scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/canvas/ScilabCanvasBridge.java
@@ -14,7 +14,7 @@ package org.scilab.modules.gui.canvas;
 
 import java.awt.image.BufferedImage;
 
-import javax.media.opengl.GL;
+import com.jogamp.opengl.GL;
 
 import org.scilab.modules.gui.bridge.canvas.SwingScilabCanvas;
 import org.scilab.modules.gui.utils.Position;
Index: scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/canvas/SimpleCanvas.java
===================================================================
--- scilab-5.5.2.orig/modules/gui/src/java/org/scilab/modules/gui/canvas/SimpleCanvas.java
+++ scilab-5.5.2/modules/gui/src/java/org/scilab/modules/gui/canvas/SimpleCanvas.java
@@ -15,7 +15,7 @@ package org.scilab.modules.gui.canvas;
 import org.scilab.modules.gui.utils.Position;
 import org.scilab.modules.gui.utils.Size;
 
-import javax.media.opengl.GL;
+import com.jogamp.opengl.GL;
 import java.awt.image.BufferedImage;
 
 /**
Index: scilab-5.5.2/modules/scirenderer/examples/org/scilab/forge/scirenderer/examples/JScrollPaneExample.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/examples/org/scilab/forge/scirenderer/examples/JScrollPaneExample.java
+++ scilab-5.5.2/modules/scirenderer/examples/org/scilab/forge/scirenderer/examples/JScrollPaneExample.java
@@ -13,8 +13,8 @@ import org.scilab.forge.scirenderer.tran
 import org.scilab.forge.scirenderer.tranformations.TransformationFactory;
 import org.scilab.forge.scirenderer.utils.shapes.geometry.CubeFactory;
 
-import javax.media.opengl.awt.GLCanvas;
-import javax.media.opengl.awt.GLJPanel;
+import com.jogamp.opengl.awt.GLCanvas;
+import com.jogamp.opengl.awt.GLJPanel;
 import javax.swing.*;
 import java.awt.*;
 
Index: scilab-5.5.2/modules/scirenderer/examples/org/scilab/forge/scirenderer/examples/utils/ExampleFrame.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/examples/org/scilab/forge/scirenderer/examples/utils/ExampleFrame.java
+++ scilab-5.5.2/modules/scirenderer/examples/org/scilab/forge/scirenderer/examples/utils/ExampleFrame.java
@@ -15,7 +15,7 @@ import com.jogamp.opengl.util.Animator;
 import org.scilab.forge.scirenderer.Canvas;
 import org.scilab.forge.scirenderer.implementation.jogl.JoGLCanvasFactory;
 
-import javax.media.opengl.awt.GLJPanel;
+import com.jogamp.opengl.awt.GLJPanel;
 import javax.swing.JFrame;
 import javax.swing.JPanel;
 import java.awt.BorderLayout;
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLCanvas.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLCanvas.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLCanvas.java
@@ -16,16 +16,16 @@ import java.awt.image.BufferedImage;
 import java.lang.reflect.InvocationTargetException;
 import java.util.concurrent.Semaphore;
 
-import javax.media.opengl.DebugGL2;
-import javax.media.opengl.GL2;
-import javax.media.opengl.GLAutoDrawable;
-import javax.media.opengl.GLCapabilities;
-import javax.media.opengl.GLContext;
-import javax.media.opengl.GLDrawableFactory;
-import javax.media.opengl.GLEventListener;
-import javax.media.opengl.GLException;
-import javax.media.opengl.GLOffscreenAutoDrawable;
-import javax.media.opengl.GLProfile;
+import com.jogamp.opengl.DebugGL2;
+import com.jogamp.opengl.GL2;
+import com.jogamp.opengl.GLAutoDrawable;
+import com.jogamp.opengl.GLCapabilities;
+import com.jogamp.opengl.GLContext;
+import com.jogamp.opengl.GLDrawableFactory;
+import com.jogamp.opengl.GLEventListener;
+import com.jogamp.opengl.GLException;
+import com.jogamp.opengl.GLOffscreenAutoDrawable;
+import com.jogamp.opengl.GLProfile;
 import javax.swing.SwingUtilities;
 
 import org.scilab.forge.scirenderer.Canvas;
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLCanvasFactory.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLCanvasFactory.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLCanvasFactory.java
@@ -13,7 +13,7 @@ package org.scilab.forge.scirenderer.imp
 
 import org.scilab.forge.scirenderer.Canvas;
 
-import javax.media.opengl.GLAutoDrawable;
+import com.jogamp.opengl.GLAutoDrawable;
 
 /**
  * @author Pierre Lando
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLCapacity.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLCapacity.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLCapacity.java
@@ -11,7 +11,7 @@
 
 package org.scilab.forge.scirenderer.implementation.jogl;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 
 /**
  * This class store current OpenGl context capacity.
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLDrawingTools.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLDrawingTools.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLDrawingTools.java
@@ -30,7 +30,7 @@ import org.scilab.forge.scirenderer.tran
 import org.scilab.forge.scirenderer.tranformations.TransformationManagerListener;
 import org.scilab.forge.scirenderer.tranformations.Vector3d;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 
 /**
  *
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLParameters.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLParameters.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/JoGLParameters.java
@@ -11,8 +11,8 @@
 
 package org.scilab.forge.scirenderer.implementation.jogl;
 
-import javax.media.opengl.GL;
-import javax.media.opengl.glu.GLU;
+import com.jogamp.opengl.GL;
+import com.jogamp.opengl.glu.GLU;
 
 /**
  * @author Pierre Lando
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLBuffersManager.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLBuffersManager.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLBuffersManager.java
@@ -16,7 +16,7 @@ import org.scilab.forge.scirenderer.buff
 import org.scilab.forge.scirenderer.buffers.ElementsBuffer;
 import org.scilab.forge.scirenderer.buffers.IndicesBuffer;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 import java.util.Collection;
 import java.util.HashSet;
 import java.util.Set;
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLDataBuffer.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLDataBuffer.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLDataBuffer.java
@@ -13,7 +13,7 @@ package org.scilab.forge.scirenderer.imp
 
 import org.scilab.forge.scirenderer.buffers.DataBuffer;
 
-import javax.media.opengl.GL;
+import com.jogamp.opengl.GL;
 import java.nio.Buffer;
 
 /**
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLElementsBuffer.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLElementsBuffer.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLElementsBuffer.java
@@ -13,7 +13,7 @@ package org.scilab.forge.scirenderer.imp
 
 import org.scilab.forge.scirenderer.buffers.ElementsBuffer;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 import java.nio.Buffer;
 import java.nio.FloatBuffer;
 
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLIndicesBuffer.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLIndicesBuffer.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/buffers/JoGLIndicesBuffer.java
@@ -13,7 +13,7 @@ package org.scilab.forge.scirenderer.imp
 
 import org.scilab.forge.scirenderer.buffers.IndicesBuffer;
 
-import javax.media.opengl.GL;
+import com.jogamp.opengl.GL;
 import java.nio.Buffer;
 import java.nio.IntBuffer;
 import java.util.Collection;
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/clipping/JoGLClippingManager.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/clipping/JoGLClippingManager.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/clipping/JoGLClippingManager.java
@@ -15,7 +15,7 @@ import org.scilab.forge.scirenderer.clip
 import org.scilab.forge.scirenderer.clipping.ClippingPlane;
 import org.scilab.forge.scirenderer.implementation.jogl.JoGLDrawingTools;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 
 /**
  * @author Pierre Lando
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/clipping/JoGLClippingPlane.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/clipping/JoGLClippingPlane.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/clipping/JoGLClippingPlane.java
@@ -17,7 +17,7 @@ import org.scilab.forge.scirenderer.tran
 import org.scilab.forge.scirenderer.tranformations.TransformationFactory;
 import org.scilab.forge.scirenderer.tranformations.Vector4d;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 
 /**
  * @author Pierre Lando
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/drawer/JoGLShapeDrawer.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/drawer/JoGLShapeDrawer.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/drawer/JoGLShapeDrawer.java
@@ -23,7 +23,7 @@ import org.scilab.forge.scirenderer.text
 import org.scilab.forge.scirenderer.lightning.LightManager;
 import org.scilab.forge.scirenderer.shapes.appearance.Material;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 import java.nio.FloatBuffer;
 import java.nio.IntBuffer;
 
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/lightning/JoGLLight.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/lightning/JoGLLight.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/lightning/JoGLLight.java
@@ -11,7 +11,7 @@
 
 package org.scilab.forge.scirenderer.implementation.jogl.lightning;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 
 import org.scilab.forge.scirenderer.implementation.jogl.utils.GLShortCuts;
 import org.scilab.forge.scirenderer.lightning.Light;
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/lightning/JoGLLightManager.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/lightning/JoGLLightManager.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/lightning/JoGLLightManager.java
@@ -17,7 +17,7 @@ import org.scilab.forge.scirenderer.ligh
 import org.scilab.forge.scirenderer.lightning.LightManager;
 import org.scilab.forge.scirenderer.shapes.appearance.Material;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 
 /**
  * JoGL implementation of {@link LightManager}
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/picking/GLPickingManager.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/picking/GLPickingManager.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/picking/GLPickingManager.java
@@ -11,7 +11,7 @@
 
 package org.scilab.forge.scirenderer.implementation.jogl.picking;
 
-import javax.media.opengl.GL;
+import com.jogamp.opengl.GL;
 
 /**
  * @author Pierre Lando
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/picking/JoGLPickingTools.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/picking/JoGLPickingTools.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/picking/JoGLPickingTools.java
@@ -16,7 +16,7 @@ import org.scilab.forge.scirenderer.impl
 import org.scilab.forge.scirenderer.picking.PickingTools;
 import org.scilab.forge.scirenderer.tranformations.Vector3d;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 import java.awt.Point;
 import java.nio.FloatBuffer;
 
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/renderer/JoGLRenderer.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/renderer/JoGLRenderer.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/renderer/JoGLRenderer.java
@@ -15,7 +15,7 @@ import org.scilab.forge.scirenderer.Draw
 import org.scilab.forge.scirenderer.implementation.jogl.JoGLDrawingTools;
 import org.scilab.forge.scirenderer.renderer.Renderer;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 
 /**
  * @author Pierre Lando
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/renderer/JoGLRendererManager.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/renderer/JoGLRendererManager.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/renderer/JoGLRendererManager.java
@@ -15,7 +15,7 @@ import org.scilab.forge.scirenderer.impl
 import org.scilab.forge.scirenderer.renderer.Renderer;
 import org.scilab.forge.scirenderer.renderer.RendererManager;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 import java.util.HashSet;
 import java.util.Set;
 import java.util.Stack;
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/texture/JoGLTextureManager.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/texture/JoGLTextureManager.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/texture/JoGLTextureManager.java
@@ -28,11 +28,11 @@ import org.scilab.forge.scirenderer.tran
 import org.scilab.forge.scirenderer.tranformations.Vector3d;
 
 
-import javax.media.opengl.GL;
-import javax.media.opengl.GL2;
-import javax.media.opengl.GL2ES1;
-import javax.media.opengl.GL2GL3;
-import javax.media.opengl.GLProfile;
+import com.jogamp.opengl.GL;
+import com.jogamp.opengl.GL2;
+import com.jogamp.opengl.GL2ES1;
+import com.jogamp.opengl.GL2GL3;
+import com.jogamp.opengl.GLProfile;
 import java.awt.Dimension;
 import java.nio.ByteBuffer;
 import java.nio.FloatBuffer;
Index: scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/utils/GLShortCuts.java
===================================================================
--- scilab-5.5.2.orig/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/utils/GLShortCuts.java
+++ scilab-5.5.2/modules/scirenderer/src/org/scilab/forge/scirenderer/implementation/jogl/utils/GLShortCuts.java
@@ -14,7 +14,7 @@ package org.scilab.forge.scirenderer.imp
 import org.scilab.forge.scirenderer.shapes.appearance.Appearance;
 import org.scilab.forge.scirenderer.shapes.appearance.Color;
 
-import javax.media.opengl.GL2;
+import com.jogamp.opengl.GL2;
 
 /**
  *

Index: scilab-5.5.2/configure
===================================================================
--- scilab-5.5.2.orig/configure
+++ scilab-5.5.2/configure
@@ -14649,7 +14649,7 @@ $as_echo_n "checking gluegen2-rt... " >&
 // #line 14649 "configure"
 import java.util.regex.Pattern;
 
-import jogamp.common.os.MachineDescriptionRuntime;
+import jogamp.common.os.MachineDataInfoRuntime;
 
 public class conftest {
     public static void main(String[] argv) {
@@ -14735,10 +14735,10 @@ $as_echo "$jar_resolved" >&6; }
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
       if test "" = "yes"; then
-         { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Could not find or use the Java package/jar gluegen2-rt used by Scilab 3D rendering (looking for package jogamp.common.os.MachineDescriptionRuntime)" >&5
-$as_echo "$as_me: WARNING: Could not find or use the Java package/jar gluegen2-rt used by Scilab 3D rendering (looking for package jogamp.common.os.MachineDescriptionRuntime)" >&2;}
+         { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Could not find or use the Java package/jar gluegen2-rt used by Scilab 3D rendering (looking for package jogamp.common.os.MachineDataInfoRuntime)" >&5
+$as_echo "$as_me: WARNING: Could not find or use the Java package/jar gluegen2-rt used by Scilab 3D rendering (looking for package jogamp.common.os.MachineDataInfoRuntime)" >&2;}
       else
-          as_fn_error $? "Could not find or use the Java package/jar gluegen2-rt used by Scilab 3D rendering (looking for package jogamp.common.os.MachineDescriptionRuntime)" "$LINENO" 5
+          as_fn_error $? "Could not find or use the Java package/jar gluegen2-rt used by Scilab 3D rendering (looking for package jogamp.common.os.MachineDataInfoRuntime)" "$LINENO" 5
       fi
     fi
 
Index: scilab-5.5.2/configure.ac
===================================================================
--- scilab-5.5.2.orig/configure.ac
+++ scilab-5.5.2/configure.ac
@@ -943,7 +943,7 @@ interface for JOGL2 - or libGL (OpenGL l
                 fi
 
                 # JoGL Native <=> Java connector
-                AC_JAVA_CHECK_PACKAGE([gluegen2-rt],[jogamp.common.os.MachineDescriptionRuntime],[Scilab 3D rendering])
+                AC_JAVA_CHECK_PACKAGE([gluegen2-rt],[jogamp.common.os.MachineDataInfoRuntime],[Scilab 3D rendering])
                 GLUEGEN2_RT=$PACKAGE_JAR_FILE
                 AC_SUBST(GLUEGEN2_RT)
 