--- modules/xcos/src/java/org/scilab/modules/xcos/Xcos.java.orig	2017-08-30 13:17:48.301380837 +0200
+++ modules/xcos/src/java/org/scilab/modules/xcos/Xcos.java	2017-08-30 13:16:58.934524972 +0200
@@ -101,7 +101,7 @@
      * Dependencies version
      */
     private static final List<String> MXGRAPH_VERSIONS = null;
-    private static final List<String> BATIK_VERSIONS = Arrays.asList("1.7", "1.8pre", "1.8");
+    private static final List<String> BATIK_VERSIONS = Arrays.asList("1.7", "1.8pre", "1.8", "1.9");
 
     private static final String UNABLE_TO_LOAD_JGRAPHX = Messages.gettext("Unable to load the jgraphx library.\nExpecting version %s ; Getting version %s .");
     private static final String UNABLE_TO_LOAD_BATIK = Messages.gettext("Unable to load the Batik library. \nExpecting version %s ; Getting version %s .");
