diff --git a/packages/python/plotly/plotly/tests/test_core/test_errors/test_dict_path_errors.py b/packages/python/plotly/plotly/tests/test_core/test_errors/test_dict_path_errors.py
index 8b899a6f5..6ee06e03f 100644
--- a/packages/python/plotly/plotly/tests/test_core/test_errors/test_dict_path_errors.py
+++ b/packages/python/plotly/plotly/tests/test_core/test_errors/test_dict_path_errors.py
@@ -402,7 +402,7 @@ template_layout_plot_bgcolor_x
         print(e.args[0])
         e_substr = error_substr(
             e.args[0],
-            """string indices must be integers
+            """string indices must be integers, not 'str'
 
 Invalid value received for the 'plot_bgcolor' property of layout
 
diff --git a/packages/python/plotly/setup.cfg b/packages/python/plotly/setup.cfg
index 1a82ac6f9..419173843 100644
--- a/packages/python/plotly/setup.cfg
+++ b/packages/python/plotly/setup.cfg
@@ -1,6 +1,6 @@
 [metadata]
 description_file = README.md
-license_file = LICENSE.txt
+license_files = LICENSE.txt
 
 [bdist_wheel]
 universal=1
diff --git a/packages/python/plotly/setup.py b/packages/python/plotly/setup.py
index 3614cb970..178f17645 100644
--- a/packages/python/plotly/setup.py
+++ b/packages/python/plotly/setup.py
@@ -495,7 +495,6 @@ versioneer_cmds = versioneer.get_cmdclass()
 setup(
     name="plotly",
     version=versioneer.get_version(),
-    use_2to3=False,
     author="Chris P",
     author_email="chris@plot.ly",
     maintainer="Nicolas Kruchten",

