diff -ruN '--exclude=build' a/include/gz/transport/detail/Node.hh b/include/gz/transport/detail/Node.hh
--- a/include/gz/transport/detail/Node.hh	2026-07-24 18:48:29.476948314 +0800
+++ b/include/gz/transport/detail/Node.hh	2026-07-24 18:48:29.488764433 +0800
@@ -31,7 +31,7 @@
         const std::string &_topic,
         const AdvertiseMessageOptions &_options)
     {
-      return this->Advertise(_topic, MessageT().GetTypeName(), _options);
+      return this->Advertise(_topic, std::string(MessageT().GetTypeName()), _options);
     }
 
     //////////////////////////////////////////////////
diff -ruN '--exclude=build' a/include/gz/transport/ReqHandler.hh b/include/gz/transport/ReqHandler.hh
--- a/include/gz/transport/ReqHandler.hh	2026-07-24 18:48:29.476614707 +0800
+++ b/include/gz/transport/ReqHandler.hh	2026-07-24 18:48:29.495537210 +0800
@@ -381,7 +381,7 @@
       public: virtual std::string ReqTypeName() const
       {
         if (this->reqMsg)
-          return this->reqMsg->GetTypeName();
+          return std::string(this->reqMsg->GetTypeName());
         else
         {
           std::cerr << "ReqHandler::ReqTypeName() Warning: Using ReqTypeName() "
@@ -394,7 +394,7 @@
       public: virtual std::string RepTypeName() const
       {
         if (this->repMsg)
-          return this->repMsg->GetTypeName();
+          return std::string(this->repMsg->GetTypeName());
         else
         {
           std::cerr << "ReqHandler::RepTypeName() Warning: Using RepTypeName() "
diff -ruN '--exclude=build' a/include/gz/transport/SubscriptionHandler.hh b/include/gz/transport/SubscriptionHandler.hh
--- a/include/gz/transport/SubscriptionHandler.hh	2026-07-24 18:48:29.476682232 +0800
+++ b/include/gz/transport/SubscriptionHandler.hh	2026-07-24 18:48:29.491010247 +0800
@@ -185,7 +185,7 @@
       // Documentation inherited.
       public: std::string TypeName()
       {
-        return T().GetTypeName();
+        return std::string(T().GetTypeName());
       }
 
       /// \brief Set the callback for this handler.
diff -ruN '--exclude=build' a/src/Node.cc b/src/Node.cc
--- a/src/Node.cc	2026-07-24 18:48:29.481424570 +0800
+++ b/src/Node.cc	2026-07-24 18:48:29.497846835 +0800
@@ -423,7 +423,7 @@
     };
 
     if (!this->dataPtr->shared->Publish(this->dataPtr->publisher.Topic(),
-          msgBuffer, msgSize, myDeallocator, _msg.GetTypeName()))
+          msgBuffer, msgSize, myDeallocator, std::string(_msg.GetTypeName())))
     {
       return false;
     }
